---
title: "Create Figures for Hicks et al. scRNA-seq paper"
output: html_document
---

```{r setup, include=FALSE}
library(knitr)
knitr::opts_chunk$set(message=FALSE, warning=FALSE)
```

# Load libraries
```{r, echo = FALSE}
library(ggplot2)
library(grid)
library(cowplot)
library(Biobase)
library(GenomicRanges)
library(stringr)
library(reshape2)
library(tidyr)
library(RColorBrewer)
library(plyr)
library(dplyr)
library(irlba)
library(rafalib) 

tidyQuants <- function(object, CDRval, qRange, filterAbove = 1){
    keepIndex <- object >= filterAbove
    out = sapply(seq_len(ncol(object)), function(x){
        quantile(object[keepIndex[,x], x], probs = qRange) })
    
    dat = data.frame("CDR" = CDRval, 
                     matrix(out, ncol = length(qRange), byrow = TRUE))
    colnames(dat)[-1] <- as.character(qRange)
    dat = gather(dat, key = CDR)
    colnames(dat)[2] <- "Quantile"
    return(dat)
}

sm <- function(x, y, x.log = FALSE,n.bins = 25){
  if(x.log){ 
    brks <- unique(quantile(x, probs = seq(0,1,len=25))) 
  } else {
    brks <- 2^unique(quantile(log2(x), probs = seq(0,1,len=n.bins))) 
  }
  mids <- (brks[-1] + brks[-length(brks)] )/ 2
  x.in <- cut(x, breaks = brks, include.lowest = TRUE)
  m <- tapply(y, x.in, mean)
  fit = lm(y~x)
  l <- predict(fit, newdata = data.frame(x  = mids))
  dat <- data.frame(x=mids, y = m, n = as.numeric(table(x.in)), pred = l)
}

gg_color_hue <- function(n) {
  hues = seq(15, 375, length=n+1)
  hcl(h=hues, l=65, c=100)[1:n]
}

working_path <- "/net/irizarryfs01/srv/export/irizarryfs01_backed_up/share_root/shicks/papers/scBatchPaper"
datapkgs_path <- "/net/irizarryfs01/srv/export/irizarryfs01/share_root/shicks/dataPackages"
list.files(working_path)

title.deng <- "Deng et al. (2014)"
title.guo <- "Guo et al. (2015)"
title.kowalczyk <- "Kowalczyk et al. (2015)"
title.kumar <- "Kumar et al. (2014)"
title.patel <- "Patel et al. (2014)"
title.treutlein <- "Treutlein et al. (2014)"
title.shalek.2014 <- "Shalek et al. (2014)"
title.trapnell <- "Trapnell et al. (2014)"
title.burns <- "Burns et al. (2015)"
title.leng <- "Leng et al. (2015)"
title.bose <- "Bose et al. (2015)"
title.jaitin <- "Jaitin et al. (2014)"
title.macosko <- "Macosko et al. (2015)"
title.satija <- "Satija et al. (2015)"
title.zeisel <- "Zeisel et al. (2015)"
```


# Load data sets

Source in one file to load data from 15 studies. The second file 
is the same as the first except it removes the `colMeans()` from 
the cells on the log2 scale to compute distances between cells. 
The third file is used to calculate the detection rate and 
the `colMeans()` from the cells on the log scale. 
```{r, load-Datasets}
# source(file.path(working_path, "loadData.r"))
# source(file.path(working_path, "loadData_colMeansRemoved.r"))
# source(file.path(working_path, "loadData_logscale.r"))
```


# Figure: detection rate boxplots by study
```{r Figure detection rate boxplots all studies, fig.width=30, fig.height=18}

scores <- rbind(data.frame("CDR" = pdDeng$CDR, "Study"= rep(title.deng, nrow(pdDeng))),
                data.frame("CDR" = pdGuo$CDR, "Study"= rep(title.guo, nrow(pdGuo))),
                data.frame("CDR" = pdKowalczyk$CDR, "Study"= rep(title.kowalczyk, nrow(pdKowalczyk))),
                data.frame("CDR" = pdKumar$CDR, "Study"= rep(title.kumar, nrow(pdKumar))),
                data.frame("CDR" = pdPatel$CDR, "Study"= rep(title.patel, nrow(pdPatel))),
                data.frame("CDR" = pdTreutlein$CDR, "Study"= rep(title.treutlein, nrow(pdTreutlein))),
                data.frame("CDR" = pdShalek$CDR, "Study"= rep(title.shalek.2014, nrow(pdShalek))),
                data.frame("CDR" = pdTrapnell$CDR, "Study"= rep(title.trapnell, nrow(pdTrapnell))),
                data.frame("CDR" = pdBurns$CDR, "Study"= rep(title.burns, nrow(pdBurns))),
                data.frame("CDR" = pdLeng$CDR, "Study"= rep(title.leng, nrow(pdLeng))),
                data.frame("CDR" = pdBose$CDR, "Study"= rep(title.bose, nrow(pdBose))),
                data.frame("CDR" = pdJaitin$CDR, "Study"= rep(title.jaitin, nrow(pdJaitin))),
                data.frame("CDR" = pdMacosko$CDR, "Study"= rep(title.macosko, nrow(pdMacosko))),
                data.frame("CDR" = pdSatija$CDR, "Study"= rep(title.satija, nrow(pdSatija))),
                data.frame("CDR" = pdZeisel$CDR, "Study"= rep(title.zeisel, nrow(pdZeisel))))

scores$Study = factor(scores$Study, 
                      levels = names(sort(unlist(lapply( split(scores$CDR, factor(scores$Study)), median)), 
                                          decreasing = FALSE)))

p.box.all <- ggplot(scores, aes(x=Study, y =CDR)) + geom_boxplot() + 
    labs(title = "Proportion of reported zeros varies across cells and studies") +
    xlab("Study") + ylab("Detection Rate") + ylim(c(0, 0.655)) + 
    theme(plot.title = element_text(size = 40),
          axis.text.y = element_text(size = 35), 
          axis.title = element_text(size = 35), 
          axis.text.x = element_text(size = 20, angle = 45, hjust = 1))
p.box.all


pdf(file.path(working_path, "scBatchFig-all15-box-CDR.pdf"), width = 25, height = 15)
print(p.box.all)
dev.off()

```


# Figure: detection rate lt1 boxplots by study
```{r Figure detection rate lt1 boxplots all studies, fig.width=30, fig.height=18}

scores <- rbind(data.frame("CDR" = pdDeng$CDRlt1, "Study"= rep(title.deng, nrow(pdDeng))),
                data.frame("CDR" = pdGuo$CDRlt1, "Study"= rep(title.guo, nrow(pdGuo))),
                data.frame("CDR" = pdKowalczyk$CDRlt1, "Study"= rep(title.kowalczyk, nrow(pdKowalczyk))),
                data.frame("CDR" = pdKumar$CDRlt1, "Study"= rep(title.kumar, nrow(pdKumar))),
                data.frame("CDR" = pdPatel$CDRlt1, "Study"= rep(title.patel, nrow(pdPatel))),
                data.frame("CDR" = pdTreutlein$CDRlt1, "Study"= rep(title.treutlein, nrow(pdTreutlein))),
                data.frame("CDR" = pdShalek$CDRlt1, "Study"= rep(title.shalek.2014, nrow(pdShalek))),
                data.frame("CDR" = pdTrapnell$CDRlt1, "Study"= rep(title.trapnell, nrow(pdTrapnell))),
                data.frame("CDR" = pdBurns$CDRlt1, "Study"= rep(title.burns, nrow(pdBurns))),
                data.frame("CDR" = pdLeng$CDRlt1, "Study"= rep(title.leng, nrow(pdLeng))),
                data.frame("CDR" = pdBose$CDRlt1, "Study"= rep(title.bose, nrow(pdBose))),
                data.frame("CDR" = pdJaitin$CDRlt1, "Study"= rep(title.jaitin, nrow(pdJaitin))),
                data.frame("CDR" = pdMacosko$CDRlt1, "Study"= rep(title.macosko, nrow(pdMacosko))),
                data.frame("CDR" = pdSatija$CDRlt1, "Study"= rep(title.satija, nrow(pdSatija))),
                data.frame("CDR" = pdZeisel$CDRlt1, "Study"= rep(title.zeisel, nrow(pdZeisel))))

scores$Study = factor(scores$Study, 
                      levels = names(sort(unlist(lapply( split(scores$CDR, factor(scores$Study)), median)), 
                                          decreasing = FALSE)))

p.box.all <- ggplot(scores, aes(x=Study, y =CDR)) + geom_boxplot() + 
    labs(title = "Proportion of reported zeros varies across cells and studies") +
    xlab("Study") + ylab("Detection Rate") + ylim(c(0, 0.655)) + 
    theme(plot.title = element_text(size = 40),
          axis.text.y = element_text(size = 35), 
          axis.title = element_text(size = 35), 
          axis.text.x = element_text(size = 20, angle = 45, hjust = 1))
p.box.all


pdf(file.path(working_path, "scBatchFig-all15-box-CDRlt1.pdf"), width = 25, height = 15)
print(p.box.all)
dev.off()

```


# Histograms to explore threshold cutoff for detection rate
```{r Figure histograms to explore cutoff for CDR EDA}
ind = 1

p.deng <- ggplot(melt(eDeng[,ind]), aes(x = value)) + geom_histogram(bins = 50) +
  geom_vline(aes(xintercept=1, color = "red"), size = 2) + 
  theme(legend.position = "none") + coord_cartesian(ylim=c(0, 1000)) + 
  xlab("Expression (log2)") + labs(title = title.deng) + 
  theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_text(size = 20))

p.guo <- ggplot(melt(eGuo[,ind]), aes(x = value)) + geom_histogram(bins = 50) + 
  geom_vline(aes(xintercept=1, color = "red"), size = 2) + 
  theme(legend.position = "none") + coord_cartesian(ylim=c(0, 1000)) + 
  xlab("Expression (log2)") + labs(title = title.guo) +
  theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_text(size = 20))
    
p.kowalczyk <- ggplot(melt(eKowalczyk[,ind]), aes(x = value)) + geom_histogram(bins = 50) +
  geom_vline(aes(xintercept=1, color = "red"), size = 2) + 
  theme(legend.position = "none") + coord_cartesian(ylim=c(0, 1000)) + 
  xlab("Expression (log2)") + labs(title = title.kowalczyk) +
  theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_text(size = 20))

p.kumar <- ggplot(melt(eKumar[,ind]), aes(x = value)) + geom_histogram(bins = 50) + 
  geom_vline(aes(xintercept=1, color = "red"), size = 2) + 
  theme(legend.position = "none") + coord_cartesian(ylim=c(0, 1000)) + 
  xlab("Expression (log2)") + labs(title = title.kumar) +
  theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_text(size = 20))

p.patel <- ggplot(melt(patel_tpm[,ind]), aes(x = value)) + geom_histogram(bins = 50) +
  geom_vline(aes(xintercept=1, color = "red"), size = 2) + 
  theme(legend.position = "none") + coord_cartesian(ylim=c(0, 1000)) + 
  xlab("Expression (log2)") + labs(title = title.patel) +
  theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_text(size = 20))
    
p.treutlein <- ggplot(melt(eTreutlein[,ind]), aes(x = value)) + geom_histogram(bins = 50) +
  geom_vline(aes(xintercept=1, color = "red"), size = 2) + 
  theme(legend.position = "none") + coord_cartesian(ylim=c(0, 1000)) + 
  xlab("Expression (log2)") + labs(title = title.treutlein) +
  theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_text(size = 20))

p.shalek <- ggplot(melt(eShalek[,ind]), aes(x = value)) + geom_histogram(bins = 50) +
  geom_vline(aes(xintercept=1, color = "red"), size = 2) + 
  theme(legend.position = "none") + coord_cartesian(ylim=c(0, 1000)) + 
  xlab("Expression (log2)") + labs(title = title.shalek.2014) +
  theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_text(size = 20))


p.trapnell <- ggplot(melt(eTrapnell[,ind]), aes(x = value)) + geom_histogram(bins = 50) +
  geom_vline(aes(xintercept=1, color = "red"), size = 2) + 
  theme(legend.position = "none") + coord_cartesian(ylim=c(0, 1000)) + 
  xlab("Expression (log2)") + labs(title = title.trapnell) +
  theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_text(size = 20))
   

p.burns <- ggplot(melt(eBurns[,ind]), aes(x = value)) + geom_histogram(bins = 50) +
  geom_vline(aes(xintercept=1, color = "red"), size = 2) + 
  theme(legend.position = "none") + coord_cartesian(ylim=c(0, 1000)) + 
  xlab("Expression (log2)") + labs(title = title.burns) +
  theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_text(size = 20))

p.leng <- ggplot(melt(eLeng[,ind]), aes(x = value)) + geom_histogram(bins = 40) +
  geom_vline(aes(xintercept=1, color = "red"), size = 2) + 
  theme(legend.position = "none") + coord_cartesian(ylim=c(0, 1000)) + 
  xlab("Expression (log2)") + labs(title = title.leng) +
  theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_text(size = 20))

p.bose <- ggplot(melt(eBose[,ind]), aes(x = value)) + geom_histogram(bins = 50) +
  geom_vline(aes(xintercept=1, color = "red"), size = 2) + 
  theme(legend.position = "none") + coord_cartesian(ylim=c(0, 1000)) + 
  xlab("Expression (log2)") + labs(title = title.bose) +
  theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_text(size = 20))

p.jaitin <- ggplot(melt(eJaitin[,ind]), aes(x = value)) + geom_histogram(bins = 50) +
  geom_vline(aes(xintercept=1, color = "red"), size = 2) + 
  theme(legend.position = "none") + coord_cartesian(ylim=c(0, 1000)) + 
  xlab("Expression (log2)") + labs(title = title.jaitin) +
  theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_text(size = 20))

p.macosko <- ggplot(melt(eMacosko[,ind]), aes(x = value)) + geom_histogram(bins = 50) +
  geom_vline(aes(xintercept=1, color = "red"), size = 2) + 
  theme(legend.position = "none") + coord_cartesian(ylim=c(0, 1000)) + 
  xlab("Expression (log2)") + labs(title = title.macosko) +
  theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_text(size = 20))

p.satija <- ggplot(melt(eSatija[,ind]), aes(x = value)) + geom_histogram(bins = 50) +
  geom_vline(aes(xintercept=1, color = "red"), size = 2) + 
  theme(legend.position = "none") + coord_cartesian(ylim=c(0, 1000)) + 
  xlab("Expression (log2)") + labs(title = title.satija) +
  theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_text(size = 20))

p.zeisel <- ggplot(melt(eZeisel[,ind]), aes(x = value)) + geom_histogram(bins = 50) +
  geom_vline(aes(xintercept=1, color = "red"), size = 2) + 
  theme(legend.position = "none") + coord_cartesian(ylim=c(0, 1000)) + 
  xlab("Expression (log2)") + labs(title = title.zeisel) +
  theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_text(size = 20))


fig.hist.all15 <- plot_grid(p.deng, p.guo, p.kowalczyk, p.kumar, p.patel,
                        p.treutlein, p.shalek, p.trapnell, p.burns, p.leng,  
                        p.bose, p.jaitin, p.macosko, p.satija, p.zeisel, 
                        labels = LETTERS[1:15], label_size = 25, ncol = 5)
fig.hist.all15

pdf(paste0(file.path(working_path, "scBatchFig-all15-hist-ind-"), ind, ".pdf"), width = 30, height = 18)
print(fig.hist.all15)
dev.off()



```


# Figure detection rate vs colMeans() with expected mean
This figure used data that does not have `colMeans()` from each cell removed 
and is on the log scale (not log2 scale). 
```{r Figure detection rate vs colMeans plots, fig.width=30, fig.height=18}
# source(file.path(working_path, "loadData_logscale.r"))

##### Figure: Deng
M <- 1.2e5
p <- pdDeng[pdDeng$bin == "Bin D", ]$CDR
scores <- data.frame(pdDeng[pdDeng$bin %in% c("Bin D"), ],
                     "predMean" = p * log( (M/ (nrow(eDeng)*p )) + 1))
PC1.Deng.D <- ggplot(scores, aes(x = CDR, y = colMeans)) + 
    geom_point(size = 2, color = "black") + 
    xlab("Detection rate") + 
    ylab("Mean Cell Expression (log scale)") + 
    labs(title = paste0("Deng et al., 2014 (Group D)")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank(), 
          legend.position="none") + 
    geom_point(aes(x = CDR, y = predMean, color = "red"), size = 2)


##### Figure: Guo
M <- 9e4
p <- pdGuo[pdGuo$weekGroup == "7-8", ]$CDR
scores <- data.frame(pdGuo[pdGuo$weekGroup == "7-8", ], 
                       "predMean" = p * log( (M/ (nrow(eGuo)*p )) + 1))
PC1.Guo.B <- ggplot(scores, aes(x = CDR, y = colMeans)) + 
    geom_point(size = 2, color = "black") + 
    xlab("Detection rate") + 
    ylab("Mean Cell Expression (log scale)") + 
    labs(title = paste0("Guo et al., 2015 (Group B)")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank(), 
          legend.position="none") + 
    geom_point(aes(x = CDR, y = predMean, color = "red"), size = 2)

##### Figure: Kowalczyk
M <- 1.5e6
p <- pdKowalczyk$CDR 
scores <- data.frame(pdKowalczyk, 
                     "predMean" = p * log( (M  / (nrow(eKowalczyk)*p )) + 1))
PC1.Kowalczyk <- ggplot(scores, aes(x = CDR, y = colMeans)) + 
    geom_point(size = 2, color = "black") + 
    xlab("Detection rate") + 
    ylab("Mean Cell Expression (log scale)") + 
    labs(title = paste0("Kowalczyk et al., 2015")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank(), 
          legend.position="none") + 
     geom_point(aes(x = CDR, y = predMean, color = "red"), size = 2)


##### Figure: Kumar
M <- 1.8e5
p <- pdKumar[pdKumar$bin == "Group A", ]$CDR 
scores <- data.frame(pdKumar[pdKumar$bin == "Group A", ], 
                     "predMean" = p * log( (M / (nrow(eKumar)*p )) + 1))
PC1.Kumar.A <- ggplot(scores, aes(x = CDR, y = colMeans)) + 
    geom_point(size = 2, color = "black") + 
    xlab("Detection rate") + 
    ylab("Mean Cell Expression (log scale)") + 
    labs(title = paste0("Kumar et al., 2014 (Group A)")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank(), 
          legend.position="none") + 
    geom_point(aes(x = CDR, y = predMean, color = "red"), size = 2)


##### Figure: Patel
ID <- patel_pd$includeSample == TRUE & patel_pd$sampleType == "SC" & 
  patel_pd$cellType == "Glioblastoma"
esetSub <- as.matrix(patel_tpm[, ID])
pdSub <- patel_pd[ID, ]
pdSub$colMeans <- colMeans(esetSub)

M <- 2.3e5
p <- pdSub$CDR 
scores <- data.frame(pdSub, 
                     "predMean" = p * log( (M / (nrow(esetSub)*p )) + 1))
PC1.Patel <- ggplot(scores, aes(x = CDR, y = colMeans)) + 
    geom_point(size = 2, color = "black") + 
    xlab("Detection rate") + 
    ylab("Mean Cell Expression (log scale)") + 
    labs(title = paste0("Patel et al., 2014")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank(), 
          legend.position="none") + 
    geom_point(aes(x = CDR, y = predMean, color = "red"), size = 2)


##### Figure: Treutlein
M <- 6e4
p <- pdTreutlein$CDR 
scores <- data.frame(pdTreutlein, 
                     "predMean" = p * log( (M / (nrow(eTreutlein)*p )) + 1))
PC1.Treutlein <- ggplot(scores, aes(x = CDR, y = colMeans)) + 
    geom_point(size = 2, color = "black") + 
    xlab("Detection rate") + 
    ylab("Mean Cell Expression (log scale)") + 
    labs(title = paste0("Treutlein et al., 2014")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank(), 
          legend.position="none") + 
    geom_point(aes(x = CDR, y = predMean, color = "red"), size = 2)


##### Figure: Shalek
M <- 1e5
p <- pdShalek$CDR 
scores <- data.frame(pdShalek, 
                      "predMean" = p * log( (M / (nrow(eShalek)*p )) + 1))
PC1.Shalek <- ggplot(scores, aes(x = CDR, y = colMeans)) + 
    geom_point(size = 2, color = "black") + 
    xlab("Detection rate") + 
    ylab("Mean Cell Expression (log scale)") + 
    labs(title = paste0("Shalek et al., 2014")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank(), 
          legend.position="none") + 
    geom_point(aes(x = CDR, y = predMean, color = "red"), size = 2)

##### Figure: Trapnell
M <- 1e5
p <- pdTrapnell$CDR 
scores <- data.frame(pdTrapnell, 
                      "predMean" = p * log( (M / (nrow(eTrapnell)*p )) + 1))
PC1.Trapnell <- ggplot(scores, aes(x = CDR, y = colMeans)) + 
    geom_point(size = 2, color = "black") + 
    xlab("Detection rate") + 
    ylab("Mean Cell Expression (log scale)") + 
    labs(title = paste0("Trapnell et al., 2014")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank(), 
          legend.position="none") + 
    geom_point(aes(x = CDR, y = predMean, color = "red"), size = 2)


##### Figure: Burns
M <- 1.8e5
p <- pdBurns[pdBurns$tissue == "cochlear", ]$CDR 
scores <- data.frame(pdBurns[pdBurns$tissue == "cochlear", ], 
                      "predMean" = p * log( (M / (nrow(eBurns)*p )) + 1))
PC1.Burns.A <- ggplot(scores, aes(x = CDR, y = colMeans)) + 
    geom_point(size = 2, color = "black") + 
    xlab("Detection rate") + 
    ylab("Mean Cell Expression (log scale)") + 
    labs(title = paste0("Burns et al., 2015 (Group A)")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank(), 
          legend.position="none") + 
    geom_point(aes(x = CDR, y = predMean, color = "red"), size = 2)


##### Figure: Leng
M <- 6e5
p <- pdLeng$CDR
scores <- data.frame(pdLeng, 
                     "predMean" = p * log( (M / (nrow(eLeng)*p )) + 1)) 
PC1.Leng <- ggplot(scores, aes(x = CDR, y = colMeans)) + 
    geom_point(size = 2, color = "black") + 
    xlab("Detection rate") + 
    ylab("Mean Cell Expression (log scale)") + 
    labs(title = paste0("Leng et al., 2015")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank(), 
          legend.position="none") + 
    geom_point(aes(x = CDR, y = predMean, color = "red"), size = 2)


##### Figure: Bose
M <- 5e5
p <- pdBose$CDR
scores <- data.frame(pdBose,
                     "predMean" = p * log( (M / (nrow(eBose)*p )) + 1)) 
PC1.Bose <- ggplot(scores, aes(x = CDR, y = colMeans)) + 
    geom_point(size = 2, color = "black") + 
    xlab("Detection rate") + 
    ylab("Mean Cell Expression (log scale)") + 
    labs(title = paste0("Bose et al., 2015")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank(), 
          legend.position="none") + 
    geom_point(aes(x = CDR, y = predMean, color = "red"), size = 2)


##### Figure: Jaitin
M <- 6e5 
p <- pdJaitin$CDR
scores <- data.frame(pdJaitin,
                     "predMean" = p * log( (M / (nrow(eJaitin)*p )) + 1))
PC1.Jaitin <- ggplot(scores, aes(x = CDR, y = colMeans)) + 
    geom_point(size = 2, color = "black") + 
    xlab("Detection rate") + 
    ylab("Mean Cell Expression (log scale)") + 
    labs(title = paste0("Jaitin et al., 2014")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank(), 
          legend.position="none") + 
    geom_point(aes(x = CDR, y = predMean, color = "red"), size = 2)


##### Figure: Macosko
M <- 5e6 
p <- pdMacosko$CDR
scores <- data.frame(pdMacosko,
                     "predMean" = p * log( (M / (nrow(eMacosko)*p )) + 1)) 
PC1.Macosko <- ggplot(scores, aes(x = CDR, y = colMeans)) + 
    geom_point(size = 2, color = "black") + 
    xlab("Detection rate") + 
    ylab("Mean Cell Expression (log scale)") + 
    labs(title = paste0("Macosko et al., 2015")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank(), 
          legend.position="none") + 
    geom_point(aes(x = CDR, y = predMean, color = "red"), size = 2)


##### Figure: Satija
M <- 4.5e5 
p <- pdSatija$CDR
scores <- data.frame(pdSatija, 
                     "predMean" = p * log( (M / (nrow(eSatija)*p )) + 1)) 
PC1.Satija <- ggplot(scores, aes(x = CDR, y = colMeans)) + 
    geom_point(size = 2, color = "black") + 
    xlab("Detection rate") + 
    ylab("Mean Cell Expression (log scale)") + 
    labs(title = paste0("Satija et al., 2015")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank(), 
          legend.position="none") + 
    geom_point(aes(x = CDR, y = predMean, color = "red"), size = 2)


##### Figure: Zeisel
M <- 5e5
p <- pdZeisel[pdZeisel$tissue == "ca1hippocampus", ]$CDR
scores <- data.frame(pdZeisel[pdZeisel$tissue == "ca1hippocampus", ],
                    "predMean" = p * log( (M / (nrow(eZeisel)*p )) + 1))
PC1.Zeisel.B <- ggplot(scores, aes(x = CDR, y = colMeans)) +
    geom_point(size = 2, color = "black") +
    xlab("Detection rate") +
    ylab("Mean Cell Expression (log scale)") +
    labs(title = paste0("Zeisel et al., 2015 (Group B)")) +
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20),
          axis.title = element_blank(),
          legend.position="none") +
     geom_point(aes(x = CDR, y = predMean, color = "red"), size = 2)


fig2.all15 <- plot_grid(PC1.Deng.D, PC1.Guo.B, PC1.Kowalczyk, PC1.Kumar.A, PC1.Patel,
                        PC1.Treutlein, PC1.Shalek, PC1.Trapnell, PC1.Burns.A, PC1.Leng,  
                        PC1.Bose, PC1.Jaitin, PC1.Macosko, PC1.Satija, PC1.Zeisel.B, 
                        labels = LETTERS[1:15], label_size = 25, ncol = 5)
fig2.all15

pdf(file.path(working_path, "scBatchFig2-all15-CDRvscolMeans-withExpectation.pdf"), width = 30, height = 18)
print(fig2.all15)
dev.off()


```

# Figure PC1 vs detection rate plots (USING detection rate lt1, not detrended)
```{r Figure PC1 vs detection rate using lt1 plots, fig.width=30, fig.height=18}

##### Figure: Deng
scores <- data.frame(pdDeng[pdDeng$bin %in% c("Bin D"), ], sDeng.D$v[, 1:3])
if(cor(scores$X1, scores$CDRlt1) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDRlt1, degree = 1))
pc1Var <- c(sDeng.D$d^2 / sum(sDeng.D$d^2))[1]
PC1.Deng.D <- ggplot(scores, aes(x = CDRlt1, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Deng et al., 2014 (Group D) (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Guo
scores <- data.frame(pdGuo[pdGuo$weekGroup == "7-8", ], sGuo.B$v[, 1:3])
if(cor(scores$X1, scores$CDRlt1) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDRlt1, degree = 1))
pc1Var <- c(sGuo.B$d^2 / sum(sGuo.B$d^2))[1]
PC1.Guo.B <- ggplot(scores, aes(x = CDRlt1, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Guo et al., 2015 (Group B) (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Kowalczyk
scores <- data.frame(pdKowalczyk, sKowalczyk$v[, 1:3])
if(cor(scores$X1, scores$CDRlt1) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDRlt1, degree = 1))
pc1Var <- c(sKowalczyk$d^2 / sum(sKowalczyk$d^2))[1]
PC1.Kowalczyk <- ggplot(scores, aes(x = CDRlt1, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Kowalczyk et al., 2015 (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Kumar
scores <- data.frame(pdKumar[pdKumar$bin == "Group A", ], sKumar.A$v[, 1:3])
if(cor(scores$X1, scores$CDRlt1) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDRlt1, degree = 1), span = 1)
pc1Var <- c(sKumar.A$d^2 / sum(sKumar.A$d^2))[1]
PC1.Kumar.A <- ggplot(scores, aes(x = CDRlt1, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 1, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Kumar et al., 2014 (Group A) (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Patel
scores <- data.frame(pdPatel, sPatel$v[, 1:5])
if(cor(scores$X1, scores$CDRlt1) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDRlt1, degree = 1), span = 1)
pc1Var <- c(sPatel$d^2 / sum(sPatel$d^2))[1]
PC1.Patel <- ggplot(scores, aes(x = CDRlt1, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 1, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Patel et al., 2014 (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Treutlein
scores <- data.frame(pdTreutlein, sTreutlein$v[, 1:5])
if(cor(scores$X1, scores$CDRlt1) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDRlt1, degree = 1))
pc1Var <- c(sTreutlein$d^2 / sum(sTreutlein$d^2))[1]
PC1.Treutlein <- ggplot(scores, aes(x = CDRlt1, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Treutlein et al., 2014 (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Shalek
scores <- data.frame(pdShalek, sShalek$v[, 1:5])
if(cor(scores$X1, scores$CDRlt1) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDRlt1, degree = 1))
pc1Var <- c(sShalek$d^2 / sum(sShalek$d^2))[1]
PC1.Shalek <- ggplot(scores, aes(x = CDRlt1, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Shalek et al., 2014 (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Trapnell
scores <- data.frame(pdTrapnell, sTrapnell$v[, 1:5])
if(cor(scores$X1, scores$CDRlt1) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDRlt1, degree = 1))
pc1Var <- c(sTrapnell$d^2 / sum(sTrapnell$d^2))[1]
PC1.Trapnell <- ggplot(scores, aes(x = CDRlt1, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Trapnell et al., 2014 (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Burns
scores <- data.frame(pdBurns[pdBurns$tissue == "cochlear", ], sBurns.A$v[, 1:3])
if(cor(scores$X1, scores$CDRlt1) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDRlt1, degree = 1))
pc1Var <- c(sBurns.A$d^2 / sum(sBurns.A$d^2))[1]
PC1.Burns.A <- ggplot(scores, aes(x = CDRlt1, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Burns et al., 2015 (Group A) (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Leng
scores <- data.frame(pdLeng, sLeng$v[, 1:3])
if(cor(scores$X1, scores$CDRlt1) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDRlt1, degree = 1))
pc1Var <- c(sLeng$d^2 / sum(sLeng$d^2))[1]
PC1.Leng <- ggplot(scores, aes(x = CDRlt1, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Leng et al., 2015 (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Bose
scores <- data.frame(pdBose, sBose$v[, 1:3])
if(cor(scores$X1, scores$CDRlt1) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDRlt1, degree = 1))
pc1Var <- c(sBose$d^2 / sum(sBose$d^2))[1]
PC1.Bose <- ggplot(scores, aes(x = CDRlt1, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Bose et al., 2015 (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Jaitin
scores <- data.frame(pdJaitin, sJaitin$v[, 1:3])
if(cor(scores$X1, scores$CDRlt1) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDRlt1, degree = 1))
pc1Var <- c(sJaitin$d^2 / sum(sJaitin$d^2))[1]
PC1.Jaitin <- ggplot(scores, aes(x = CDRlt1, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Jaitin et al., 2014 (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Macosko
scores <- data.frame(pdMacosko, sMacosko$v[, 1:3])
if(cor(scores$X1, scores$CDRlt1) < 0){ scores$X1 <- scores$X1*(-1) } 
scores <- scores[!(scores$CDRlt1 < 0.008 & scores$X1 > 0.002), ]
fit.lo <- loess(scores$X1 ~ poly(scores$CDRlt1, degree = 1), span = 0.15)
pc1Var <- c(sMacosko$d^2 / sum(sMacosko$d^2))[1]
PC1.Macosko <- ggplot(scores, aes(x = CDRlt1, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.15, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Macosko et al., 2015 (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Satija
scores <- data.frame(pdSatija, sSatija$v[, 1:3])
if(cor(scores$X1, scores$CDRlt1) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDRlt1, degree = 1))
pc1Var <- c(sSatija$d^2 / sum(sSatija$d^2))[1]
PC1.Satija <- ggplot(scores, aes(x = CDRlt1, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Satija et al., 2015 (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Zeisel
scores <- data.frame(pdZeisel[pdZeisel$tissue == "ca1hippocampus", ], sZeisel.B$v[, 1:3])
if(cor(scores$X1, scores$CDRlt1) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDRlt1, degree = 1))
pc1Var <- c(sZeisel.B$d^2 / sum(sZeisel.B$d^2))[1]
PC1.Zeisel.B <- ggplot(scores, aes(x = CDRlt1, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Zeisel et al., 2015 (Group B) (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


fig2.all15 <- plot_grid(PC1.Deng.D, PC1.Guo.B, PC1.Kowalczyk, PC1.Kumar.A, PC1.Patel,
                        PC1.Treutlein, PC1.Shalek, PC1.Trapnell, PC1.Burns.A, PC1.Leng,  
                        PC1.Bose, PC1.Jaitin, PC1.Macosko, PC1.Satija, PC1.Zeisel.B, 
                        labels = LETTERS[1:15], label_size = 25, ncol = 5)
fig2.all15

pdf(file.path(working_path, "scBatchFig2-all15-CDRlt1.pdf"), width = 30, height = 18)
print(fig2.all15)
dev.off()

```


# Figure PC1 vs detection rate plots (USING detection rate lt1, detrended)
```{r Figure PC1 vs detection rate using lt1 detrended plots, fig.width=30, fig.height=18}

##### Figure: Deng
scores <- data.frame(pdDeng[pdDeng$bin %in% c("Bin D"), ], sDeng.D$v[, 1:3])
if(cor(scores$X1, scores$CDRlt1) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDRlt1, degree = 1), span = 1)
pc1Var <- c(sDeng.D$d^2 / sum(sDeng.D$d^2))[1]
PC1.Deng.D <- ggplot(scores, aes(x = CDRlt1, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 1, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Deng et al., 2014 (Group D) (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Guo
scores <- data.frame(pdGuo[pdGuo$weekGroup == "7-8", ], sGuo.B$v[, 1:3])
if(cor(scores$X1, scores$CDRlt1) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDRlt1, degree = 1))
pc1Var <- c(sGuo.B$d^2 / sum(sGuo.B$d^2))[1]
PC1.Guo.B <- ggplot(scores, aes(x = CDRlt1, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Guo et al., 2015 (Group B) (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Kowalczyk
scores <- data.frame(pdKowalczyk, sKowalczyk$v[, 1:3])
if(cor(scores$X1, scores$CDRlt1) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDRlt1, degree = 1))
pc1Var <- c(sKowalczyk$d^2 / sum(sKowalczyk$d^2))[1]
PC1.Kowalczyk <- ggplot(scores, aes(x = CDRlt1, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Kowalczyk et al., 2015 (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Kumar
scores <- data.frame(pdKumar[pdKumar$bin == "Group A", ], sKumar.A$v[, 1:3])
if(cor(scores$X1, scores$CDRlt1) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDRlt1, degree = 1), span = 1)
pc1Var <- c(sKumar.A$d^2 / sum(sKumar.A$d^2))[1]
PC1.Kumar.A <- ggplot(scores, aes(x = CDRlt1, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 1, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Kumar et al., 2014 (Group A) (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Patel
scores <- data.frame(pdPatel, sPatel$v[, 1:5])
if(cor(scores$X1, scores$CDRlt1) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDRlt1, degree = 1), span = 1)
pc1Var <- c(sPatel$d^2 / sum(sPatel$d^2))[1]
PC1.Patel <- ggplot(scores, aes(x = CDRlt1, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 1, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Patel et al., 2014 (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Treutlein
scores <- data.frame(pdTreutlein, sTreutlein$v[, 1:5])
if(cor(scores$X1, scores$CDRlt1) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDRlt1, degree = 1))
pc1Var <- c(sTreutlein$d^2 / sum(sTreutlein$d^2))[1]
PC1.Treutlein <- ggplot(scores, aes(x = CDRlt1, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Treutlein et al., 2014 (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Shalek
scores <- data.frame(pdShalek, sShalek$v[, 1:5])
if(cor(scores$X1, scores$CDRlt1) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDRlt1, degree = 1))
pc1Var <- c(sShalek$d^2 / sum(sShalek$d^2))[1]
PC1.Shalek <- ggplot(scores, aes(x = CDRlt1, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Shalek et al., 2014 (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Trapnell
scores <- data.frame(pdTrapnell, sTrapnell$v[, 1:5])
if(cor(scores$X1, scores$CDRlt1) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDRlt1, degree = 1))
pc1Var <- c(sTrapnell$d^2 / sum(sTrapnell$d^2))[1]
PC1.Trapnell <- ggplot(scores, aes(x = CDRlt1, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Trapnell et al., 2014 (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Burns
scores <- data.frame(pdBurns[pdBurns$tissue == "cochlear", ], sBurns.A$v[, 1:3])
if(cor(scores$X1, scores$CDRlt1) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDRlt1, degree = 1), span = 1)
pc1Var <- c(sBurns.A$d^2 / sum(sBurns.A$d^2))[1]
PC1.Burns.A <- ggplot(scores, aes(x = CDRlt1, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 1, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Burns et al., 2015 (Group A) (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Leng
scores <- data.frame(pdLeng, sLeng$v[, 1:3])
if(cor(scores$X1, scores$CDRlt1) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDRlt1, degree = 1), span = 0.95)
pc1Var <- c(sLeng$d^2 / sum(sLeng$d^2))[1]
PC1.Leng <- ggplot(scores, aes(x = CDRlt1, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.95, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Leng et al., 2015 (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Bose
scores <- data.frame(pdBose, sBose$v[, 1:3])
if(cor(scores$X1, scores$CDRlt1) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDRlt1, degree = 1))
pc1Var <- c(sBose$d^2 / sum(sBose$d^2))[1]
PC1.Bose <- ggplot(scores, aes(x = CDRlt1, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Bose et al., 2015 (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Jaitin
scores <- data.frame(pdJaitin, sJaitin$v[, 1:3])
if(cor(scores$X1, scores$CDRlt1) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDRlt1, degree = 1))
pc1Var <- c(sJaitin$d^2 / sum(sJaitin$d^2))[1]
PC1.Jaitin <- ggplot(scores, aes(x = CDRlt1, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Jaitin et al., 2014 (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Macosko
scores <- data.frame(pdMacosko, sMacosko$v[, 1:3])
if(cor(scores$X1, scores$CDRlt1) < 0){ scores$X1 <- scores$X1*(-1) } 
scores <- scores[!(scores$CDRlt1 < 0.008 & scores$X1 > 0.002), ]
fit.lo <- loess(scores$X1 ~ poly(scores$CDRlt1, degree = 1), span = 0.15)
pc1Var <- c(sMacosko$d^2 / sum(sMacosko$d^2))[1]
PC1.Macosko <- ggplot(scores, aes(x = CDRlt1, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.15, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Macosko et al., 2015 (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Satija
scores <- data.frame(pdSatija, sSatija$v[, 1:3])
if(cor(scores$X1, scores$CDRlt1) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDRlt1, degree = 1))
pc1Var <- c(sSatija$d^2 / sum(sSatija$d^2))[1]
PC1.Satija <- ggplot(scores, aes(x = CDRlt1, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Satija et al., 2015 (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Zeisel
scores <- data.frame(pdZeisel[pdZeisel$tissue == "ca1hippocampus", ], sZeisel.B$v[, 1:3])
if(cor(scores$X1, scores$CDRlt1) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDRlt1, degree = 1))
pc1Var <- c(sZeisel.B$d^2 / sum(sZeisel.B$d^2))[1]
PC1.Zeisel.B <- ggplot(scores, aes(x = CDRlt1, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Zeisel et al., 2015 (Group B) (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


fig2.all15 <- plot_grid(PC1.Deng.D, PC1.Guo.B, PC1.Kowalczyk, PC1.Kumar.A, PC1.Patel,
                        PC1.Treutlein, PC1.Shalek, PC1.Trapnell, PC1.Burns.A, PC1.Leng,  
                        PC1.Bose, PC1.Jaitin, PC1.Macosko, PC1.Satija, PC1.Zeisel.B, 
                        labels = LETTERS[1:15], label_size = 25, ncol = 5)
fig2.all15

pdf(file.path(working_path, "scBatchFig2-all15-CDRlt1-detrended.pdf"), width = 30, height = 18)
print(fig2.all15)
dev.off()

```



# Figure PC1 vs detection rate plots (USING detection rate, not detrended)
```{r Figure PC1 vs detection rate plots, fig.width=30, fig.height=18}
##### Figure: Deng
scores <- data.frame(pdDeng[pdDeng$bin %in% c("Bin D"), ], sDeng.D$v[, 1:3])
scores <- scores[scores$CDR < 0.5, ]
if(cor(scores$X1, scores$CDR) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDR, degree = 1), span = 1)
pc1Var <- c(sDeng.D$d^2 / sum(sDeng.D$d^2))[1]
PC1.Deng.D <- ggplot(scores, aes(x = CDR, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 1, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Deng et al., 2014 (Group D) (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Guo
scores <- data.frame(pdGuo[pdGuo$weekGroup == "7-8", ], sGuo.B$v[, 1:3])
if(cor(scores$X1, scores$CDR) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDR, degree = 1))
pc1Var <- c(sGuo.B$d^2 / sum(sGuo.B$d^2))[1]
PC1.Guo.B <- ggplot(scores, aes(x = CDR, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Guo et al., 2015 (Group B) (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())

##### Figure: Kowalczyk
scores <- data.frame(pdKowalczyk, sKowalczyk$v[, 1:3])
if(cor(scores$X1, scores$CDR) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDR, degree = 1))
pc1Var <- c(sKowalczyk$d^2 / sum(sKowalczyk$d^2))[1]
PC1.Kowalczyk <- ggplot(scores, aes(x = CDR, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Kowalczyk et al., 2015 (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Kumar
scores <- data.frame(pdKumar[pdKumar$bin == "Group A", ], sKumar.A$v[, 1:3])
if(cor(scores$X1, scores$CDR) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDR, degree = 1), span = 1)
pc1Var <- c(sKumar.A$d^2 / sum(sKumar.A$d^2))[1]
PC1.Kumar.A <- ggplot(scores, aes(x = CDR, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 1, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Kumar et al., 2014 (Group A) (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Patel
scores <- data.frame(pdPatel, sPatel$v[, 1:5])
if(cor(scores$X1, scores$CDR) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDR, degree = 1), span = 1)
pc1Var <- c(sPatel$d^2 / sum(sPatel$d^2))[1]
PC1.Patel <- ggplot(scores, aes(x = CDR, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 1, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Patel et al., 2014 (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Treutlein
scores <- data.frame(pdTreutlein, sTreutlein$v[, 1:5])
if(cor(scores$X1, scores$CDR) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDR, degree = 1))
pc1Var <- c(sTreutlein$d^2 / sum(sTreutlein$d^2))[1]
PC1.Treutlein <- ggplot(scores, aes(x = CDR, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Treutlein et al., 2014 (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Shalek
scores <- data.frame(pdShalek, sShalek$v[, 1:5])
if(cor(scores$X1, scores$CDR) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDR, degree = 1))
pc1Var <- c(sShalek$d^2 / sum(sShalek$d^2))[1]
PC1.Shalek <- ggplot(scores, aes(x = CDR, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Shalek et al., 2014 (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Trapnell
scores <- data.frame(pdTrapnell, sTrapnell$v[, 1:5])
if(cor(scores$X1, scores$CDR) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDR, degree = 1))
pc1Var <- c(sTrapnell$d^2 / sum(sTrapnell$d^2))[1]
PC1.Trapnell <- ggplot(scores, aes(x = CDR, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Trapnell et al., 2014 (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Burns
scores <- data.frame(pdBurns[pdBurns$tissue == "cochlear", ], sBurns.A$v[, 1:3])
if(cor(scores$X1, scores$CDR) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDR, degree = 1))
pc1Var <- c(sBurns.A$d^2 / sum(sBurns.A$d^2))[1]
PC1.Burns.A <- ggplot(scores, aes(x = CDR, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Burns et al., 2015 (Group A) (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Leng
scores <- data.frame(pdLeng, sLeng$v[, 1:3])
if(cor(scores$X1, scores$CDR) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDR, degree = 1), span = 1)
pc1Var <- c(sLeng$d^2 / sum(sLeng$d^2))[1]
PC1.Leng <- ggplot(scores, aes(x = CDR, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 1, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Leng et al., 2015 (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Bose
scores <- data.frame(pdBose, sBose$v[, 1:3])
if(cor(scores$X1, scores$CDR) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDR, degree = 1))
pc1Var <- c(sBose$d^2 / sum(sBose$d^2))[1]
PC1.Bose <- ggplot(scores, aes(x = CDR, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Bose et al., 2015 (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Jaitin
scores <- data.frame(pdJaitin, sJaitin$v[, 1:3])
if(cor(scores$X1, scores$CDR) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDR, degree = 1))
pc1Var <- c(sJaitin$d^2 / sum(sJaitin$d^2))[1]
PC1.Jaitin <- ggplot(scores, aes(x = CDR, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Jaitin et al., 2014 (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Macosko
scores <- data.frame(pdMacosko, sMacosko$v[, 1:3])
if(cor(scores$X1, scores$CDR) < 0){ scores$X1 <- scores$X1*(-1) } 
scores <- scores[!(scores$CDR < 0.008 & scores$X1 > 0.002), ]
fit.lo <- loess(scores$X1 ~ poly(scores$CDR, degree = 1), span = 0.15)
pc1Var <- c(sMacosko$d^2 / sum(sMacosko$d^2))[1]
PC1.Macosko <- ggplot(scores, aes(x = CDR, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.15, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Macosko et al., 2015 (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Satija
scores <- data.frame(pdSatija, sSatija$v[, 1:3])
if(cor(scores$X1, scores$CDR) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDR, degree = 1))
pc1Var <- c(sSatija$d^2 / sum(sSatija$d^2))[1]
PC1.Satija <- ggplot(scores, aes(x = CDR, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Satija et al., 2015 (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Zeisel
scores <- data.frame(pdZeisel[pdZeisel$tissue == "ca1hippocampus", ], sZeisel.B$v[, 1:3])
if(cor(scores$X1, scores$CDR) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDR, degree = 1))
pc1Var <- c(sZeisel.B$d^2 / sum(sZeisel.B$d^2))[1]
PC1.Zeisel.B <- ggplot(scores, aes(x = CDR, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Zeisel et al., 2015 (Group B) (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


fig2.all15 <- plot_grid(PC1.Deng.D, PC1.Guo.B, PC1.Kowalczyk, PC1.Kumar.A, PC1.Patel,
                        PC1.Treutlein, PC1.Shalek, PC1.Trapnell, PC1.Burns.A, PC1.Leng,  
                        PC1.Bose, PC1.Jaitin, PC1.Macosko, PC1.Satija, PC1.Zeisel.B, 
                        labels = LETTERS[1:15], label_size = 25, ncol = 5)
fig2.all15

pdf(file.path(working_path, "scBatchFig2-all15-CDR.pdf"), width = 30, height = 18)
print(fig2.all15)
dev.off()

```

# Figure PC1 vs detection rate plots (USING detection rate, detrended)
```{r Figure PC1 vs detection rate detrended plots, fig.width=30, fig.height=18}
##### Figure: Deng
scores <- data.frame(pdDeng[pdDeng$bin %in% c("Bin D"), ], sDeng.D$v[, 1:3])
scores <- scores[scores$CDR < 0.5, ]
if(cor(scores$X1, scores$CDR) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDR, degree = 1), span = 1)
pc1Var <- c(sDeng.D$d^2 / sum(sDeng.D$d^2))[1]
PC1.Deng.D <- ggplot(scores, aes(x = CDR, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 1, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Deng et al., 2014 (Group D) (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Guo
scores <- data.frame(pdGuo[pdGuo$weekGroup == "7-8", ], sGuo.B$v[, 1:3])
if(cor(scores$X1, scores$CDR) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDR, degree = 1), span = 1)
pc1Var <- c(sGuo.B$d^2 / sum(sGuo.B$d^2))[1]
PC1.Guo.B <- ggplot(scores, aes(x = CDR, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 1, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Guo et al., 2015 (Group B) (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Kowalczyk
scores <- data.frame(pdKowalczyk, sKowalczyk$v[, 1:3])
if(cor(scores$X1, scores$CDR) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDR, degree = 1), span = 0.8)
pc1Var <- c(sKowalczyk$d^2 / sum(sKowalczyk$d^2))[1]
PC1.Kowalczyk <- ggplot(scores, aes(x = CDR, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.8, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Kowalczyk et al., 2015 (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Kumar
scores <- data.frame(pdKumar[pdKumar$bin == "Group A", ], sKumar.A$v[, 1:3])
if(cor(scores$X1, scores$CDR) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDR, degree = 1), span = 1)
pc1Var <- c(sKumar.A$d^2 / sum(sKumar.A$d^2))[1]
PC1.Kumar.A <- ggplot(scores, aes(x = CDR, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 1, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Kumar et al., 2014 (Group A) (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Patel
scores <- data.frame(pdPatel, sPatel$v[, 1:5])
if(cor(scores$X1, scores$CDR) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDR, degree = 1), span = 1)
pc1Var <- c(sPatel$d^2 / sum(sPatel$d^2))[1]
PC1.Patel <- ggplot(scores, aes(x = CDR, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 1, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Patel et al., 2014 (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Treutlein
scores <- data.frame(pdTreutlein, sTreutlein$v[, 1:5])
if(cor(scores$X1, scores$CDR) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDR, degree = 1))
pc1Var <- c(sTreutlein$d^2 / sum(sTreutlein$d^2))[1]
PC1.Treutlein <- ggplot(scores, aes(x = CDR, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Treutlein et al., 2014 (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Shalek
scores <- data.frame(pdShalek, sShalek$v[, 1:5])
if(cor(scores$X1, scores$CDR) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDR, degree = 1))
pc1Var <- c(sShalek$d^2 / sum(sShalek$d^2))[1]
PC1.Shalek <- ggplot(scores, aes(x = CDR, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Shalek et al., 2014 (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Trapnell
scores <- data.frame(pdTrapnell, sTrapnell$v[, 1:5])
if(cor(scores$X1, scores$CDR) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDR, degree = 1))
pc1Var <- c(sTrapnell$d^2 / sum(sTrapnell$d^2))[1]
PC1.Trapnell <- ggplot(scores, aes(x = CDR, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Trapnell et al., 2014 (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Burns
scores <- data.frame(pdBurns[pdBurns$tissue == "cochlear", ], sBurns.A$v[, 1:3])
if(cor(scores$X1, scores$CDR) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDR, degree = 1), span = 1)
pc1Var <- c(sBurns.A$d^2 / sum(sBurns.A$d^2))[1]
PC1.Burns.A <- ggplot(scores, aes(x = CDR, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 1, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Burns et al., 2015 (Group A) (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Leng
scores <- data.frame(pdLeng, sLeng$v[, 1:3])
if(cor(scores$X1, scores$CDR) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDR, degree = 1), span = 1)
pc1Var <- c(sLeng$d^2 / sum(sLeng$d^2))[1]
PC1.Leng <- ggplot(scores, aes(x = CDR, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 1, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Leng et al., 2015 (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Bose
scores <- data.frame(pdBose, sBose$v[, 1:3])
if(cor(scores$X1, scores$CDR) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDR, degree = 1))
pc1Var <- c(sBose$d^2 / sum(sBose$d^2))[1]
PC1.Bose <- ggplot(scores, aes(x = CDR, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Bose et al., 2015 (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Jaitin
scores <- data.frame(pdJaitin, sJaitin$v[, 1:3])
if(cor(scores$X1, scores$CDR) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDR, degree = 1))
pc1Var <- c(sJaitin$d^2 / sum(sJaitin$d^2))[1]
PC1.Jaitin <- ggplot(scores, aes(x = CDR, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Jaitin et al., 2014 (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Macosko
scores <- data.frame(pdMacosko, sMacosko$v[, 1:3])
if(cor(scores$X1, scores$CDR) < 0){ scores$X1 <- scores$X1*(-1) } 
scores <- scores[!(scores$CDR < 0.008 & scores$X1 > 0.002), ]
fit.lo <- loess(scores$X1 ~ poly(scores$CDR, degree = 1), span = 0.15)
pc1Var <- c(sMacosko$d^2 / sum(sMacosko$d^2))[1]
PC1.Macosko <- ggplot(scores, aes(x = CDR, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.15, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Macosko et al., 2015 (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Satija
scores <- data.frame(pdSatija, sSatija$v[, 1:3])
if(cor(scores$X1, scores$CDR) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDR, degree = 1))
pc1Var <- c(sSatija$d^2 / sum(sSatija$d^2))[1]
PC1.Satija <- ggplot(scores, aes(x = CDR, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Satija et al., 2015 (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


##### Figure: Zeisel
scores <- data.frame(pdZeisel[pdZeisel$tissue == "ca1hippocampus", ], sZeisel.B$v[, 1:3])
if(cor(scores$X1, scores$CDR) < 0){ scores$X1 <- scores$X1*(-1) } 
fit.lo <- loess(scores$X1 ~ poly(scores$CDR, degree = 1))
pc1Var <- c(sZeisel.B$d^2 / sum(sZeisel.B$d^2))[1]
PC1.Zeisel.B <- ggplot(scores, aes(x = CDR, y = X1)) + 
    geom_point(size = 2, color = "black") + 
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    ylab(paste0("Principal Component 1 (", round(pc1Var,2)*100, "%)")) + 
    xlab("Detection rate") + 
    labs(title = paste0("Zeisel et al., 2015 (Group B) (R^2: ", 
                        round(1 - var(fit.lo$residuals)/var(scores$X1), 2), ")")) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank())


fig2.all15 <- plot_grid(PC1.Deng.D, PC1.Guo.B, PC1.Kowalczyk, PC1.Kumar.A, PC1.Patel,
                        PC1.Treutlein, PC1.Shalek, PC1.Trapnell, PC1.Burns.A, PC1.Leng,  
                        PC1.Bose, PC1.Jaitin, PC1.Macosko, PC1.Satija, PC1.Zeisel.B, 
                        labels = LETTERS[1:15], label_size = 25, ncol = 5)
fig2.all15

pdf(file.path(working_path, "scBatchFig2-all15-CDR-detrended.pdf"), width = 30, height = 18)
print(fig2.all15)
dev.off()

```


# Figure: Patel et al. (2014) tumor MGH26 data 
The analysis below only uses cell from one tumor that were 
processed in two batches. The data used removed the colMeans 
from the cells before calculating PCs
```{r Supp Figure Patel Tumor 5 only, fig.width=12, fig.height=10}
# ePatel <- sweep(ePatel, 2, colMeans(ePatel), FUN = "-") # remove colmeans

pdSub <- pdPatel[pdPatel$tumorName == "MGH26", ]
p.boxplot <- ggplot(as.data.frame(pdSub), aes(x=instrument, y = CDRlt1, fill = instrument)) + 
    geom_boxplot() + ylim(0.08, 0.27) + 
    xlab("Sequencing Instrument Name") + ylab("Detection rate") +
      theme(axis.text.x = element_text(size = 30), 
            axis.text.y = element_text(size = 28),
            axis.title = element_text(size = 30),
           legend.position="none")
p.boxplot

subIDs <- pdPatel$tumorName == "MGH26"
s <- svd(ePatel[,subIDs] - rowMeans(ePatel[,subIDs]))
pcVar <- c(s$d^2 / sum(s$d^2))[1:3]
scores.sub <- data.frame(pdSub, s$v[,1:3])
if(cor(scores.sub$X1, scores.sub$CDRlt1) < 0){ scores.sub$X1 <- scores.sub$X1*(-1) } 

p.pc <- ggplot(scores.sub, aes(x=X1, y = X2, color = instrument)) + geom_point(size =5) + 
           xlab(paste0("Principal Component 1 (", round(pcVar[1],2)*100, "%)")) + 
           ylab(paste0("Principal Component 2 (", round(pcVar[2],2)*100, "%)")) + 
        theme(axis.text = element_text(size = 30), 
          axis.title = element_text(size = 30),
          legend.position="none") 

p.CDRlt1vsPC <- ggplot(scores.sub, aes(x=CDRlt1, y = X1, color = instrument)) + 
           geom_point(size = 5) + 
           xlab("Detection rate") + 
           ylab(paste0("Principal Component 1 (", round(pcVar[1],2)*100, "%)"))  + 
      theme(axis.text = element_text(size = 30), 
          axis.title.x = element_text(size = 30),
          axis.title.y = element_text(size = 30),
          legend.position="none") 


p.out <- plot_grid(p.pc,  p.CDRlt1vsPC, p.boxplot, labels = LETTERS[1:4], ncol = 3, label_size = 45)
pdf(file.path(working_path, "scBatchSuppFig-Patel-box-tumor5only.pdf"), width = 30, height = 8)
print(p.out)
dev.off()


# logistic curves 
ID <- patel_pd$includeSample == TRUE &
  patel_pd$cellType == "Glioblastoma"
esetSub <- as.matrix(patel_tpm[, ID])
pdSub <- patel_pd[ID, ]

esetBulk <- esetSub[, pdSub$sampleType == "bulk"]
esetSC <- esetSub[, pdSub$sampleType == "SC"]
pdBulk <- pdSub[pdSub$sampleType == "bulk", ]
pdSC <- pdSub[pdSub$sampleType == "SC", ]

esetBulk <- esetBulk[, pdBulk$tumorName == "MGH26"]
pdBulk <- pdBulk[ pdBulk$tumorName == "MGH26", ]
esetSC <- esetSC[, pdSC$tumorName == "MGH26"]
pdSC <- pdSC[ pdSC$tumorName == "MGH26", ]


xx = log2(esetBulk + 1) # already on log2(x+1) scale
yy = log2(esetSC + 1)

z <- 2^(seq(-3,8, by= 0.1))
est.coeffs <- array(0, dim = c(dim(yy)[2], 2))
pred.y = array(0, dim = c(length(z), dim(yy)[2]))
for(i in 1:dim(yy)[2]){
  zz <- ifelse(esetSC[,i] > 1, 1, 0)
  fit = glm(zz ~ xx, family = binomial)
  est.coeffs[i,] <- fit$coefficients
  pred.y[,i] <- locfit::expit(fit$coefficients[1] + fit$coefficients[2]*log2(z))
  print(i)
}

colnames(pred.y) <- pdSC$cell
dat <- data.frame(melt(pred.y[, c(54:118, 1:53)]), 
                  "instrument" = factor(rep(pdSC$instrument[c(54:118, 1:53)], each = length(z))), 
                  "bulk" = rep(log2(z), ncol(pred.y[, c(54:118, 1:53)])))

p.probdetect <- ggplot(dat, aes(x = bulk, y = value, group = Var2)) + 
  geom_line(aes(color = instrument), size = 1) + ylim(c(0, 1)) + 
  xlab("Bulk profile (log2 scale)") + ylab("Probability of detection") + 
     theme(axis.text = element_text(size = 25), 
          axis.title.x = element_text(size = 25),
          axis.title.y = element_text(size = 25),
          legend.text=element_text(size=20), 
          legend.title=element_text(size=20), 
          legend.position="top") + 
    scale_color_discrete(name = "Instrument")


dat2 <- data.frame(pdSC, "intercept" = est.coeffs[, 1], "slope" = est.coeffs[,2])
p.intercept <- ggplot(dat2, aes(x = CDRlt1, y = intercept, color = instrument))  + geom_point(size = 3) + 
       xlab("Detection rate") + ylab("Intercept\n(logistic regression model)") + 
       theme(axis.text = element_text(size = 25), 
          axis.title.x = element_text(size = 25),
          axis.title.y = element_text(size = 25),  
          legend.text=element_text(size=20), 
          legend.title=element_text(size=20), 
          legend.position="top") + 
        scale_color_discrete(name = "Instrument")

p.out <- plot_grid(p.probdetect, p.intercept, labels = LETTERS[1:2], ncol = 2, label_size = 25)
pdf(file.path(working_path, "scBatchSuppFig-Patel-logistic-tumor5only.pdf"), width = 15, height = 6)
print(p.out)
dev.off()


library(rafalib)
hc <- hclust(dist(t(ePatel[,subIDs])))

pdf(file.path(working_path, "scBatchSuppFig-Patel-hclust-tumor5only.pdf"), width = 15, height = 6)
myplclust(hc, labels=scores.sub$instrument, 
          lab.col=as.fumeric(scores.sub$instrument), cex=0.75, 
          main = "Cells from same tumor cluster based on sequencing instrument")
dev.off()


```


# Patel and Trapnell studies: detection rate vs sequencing depth
Patel et al. (2014) counts came from [GitHub repo](https://github.com/willtownes/patel2014gliohuman) 
and the Trapnell et al. (2014) counts came from the [conquer database](http://imlspenticton.uzh.ch:3838/conquer/). 
```{r patel and trapnell counts vs seq depth}
# Patel 
data(patel2014gliohuman_counts) # object name: patel_glio_2014
patel_pd$cS = colSums(as.data.frame(as.matrix(assay(patel_glio_2014)))) / 1e6

ID <- patel_pd$includeSample == TRUE & patel_pd$sampleType == "SC" & 
  patel_pd$cellType == "Glioblastoma"
esetSub <- as.matrix(patel_tpm[, ID])
pdSub <- patel_pd[ID, ]



# Trapnell
suppressPackageStartupMessages(library(SummarizedExperiment))
suppressPackageStartupMessages(library(MultiAssayExperiment))

conquer_data_path <- "/net/irizarryfs01/srv/export/irizarryfs01/share_root/shicks/conquer/data"
conquer_analysis_path <- "/net/irizarryfs01/srv/export/irizarryfs01_backed_up/share_root/shicks/data/conquer"
loadFiles <- list.files(conquer_data_path)
loadFiles <- c("GSE52529-GPL11154.rds", "GSE52529-GPL16791.rds") # two Trapnell files
i = 1
gseObject1 <- readRDS(file.path(conquer_data_path, loadFiles[i]))
gseObject_gene <- experiments(gseObject1)[["gene"]]
eset <- assays(gseObject_gene)[["count"]]
libSize <- colSums(eset)/1e6
eset <- assays(gseObject_gene)[["TPM"]]
CDR <- colMeans(eset > 1)


i = 2
gseObject2 <- readRDS(file.path(conquer_data_path, loadFiles[i]))
gseObject_gene <- experiments(gseObject2)[["gene"]]
eset <- assays(gseObject_gene)[["count"]]
libSize <- c(libSize, colSums(eset)/1e6)
eset <- assays(gseObject_gene)[["TPM"]]
CDR <- c(CDR, colMeans(eset > 1))
 
pd <- rbind(pData(gseObject1), pData(gseObject2))
pd$CDR <- CDR
pd$libSize <- libSize

  
pdf(file.path(working_path, "scBatchSuppFig-Patel-libSizeCDR.pdf"), width = 12, height = 6)
par(mfrow=c(1,2)) 
plot(pdSub$cS, pdSub$CDR, xlab = "Sequencing depth", 
     ylab = "Detection rate", main = "Patel et al. (2014)", pch = 16)
plot(pd$libSize, pd$CDR, xlab = "Sequencing depth", 
     ylab = "Detection rate", main = "Trapnell et al. (2014)", pch = 16)
dev.off()







```


# Figure: K-boxplots 

```{r kaboxplots, fig.width=12, fig.height=16, message=FALSE, warning=FALSE}
qVals = c(0.25, 0.5, 0.75)

dat <- tidyQuants(eDeng[, pdDeng$bin == "Bin A"], 
                  CDRval = pdDeng$CDRlt1[pdDeng$bin == "Bin A"], 
                  qRange = qVals)
pMed.Deng.A <- ggplot(dat, aes(x = CDR, y = value, colour = Quantile)) +
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    geom_point(size = 2) + 
    xlab("Detection rate") + 
    ylab("Expression (log2)") + 
    labs(title = paste0(title.deng, " (Group A)")) + 
    theme(plot.title = element_text(size =20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank(), 
          legend.position = "none")

dat <- tidyQuants(eDeng[, pdDeng$bin == "Bin B"], 
                  CDRval = pdDeng$CDRlt1[pdDeng$bin == "Bin B"], 
                  qRange = qVals)
pMed.Deng.B <- ggplot(dat, aes(x = CDR, y = value, colour = Quantile)) +
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    geom_point(size = 2) + 
    xlab("Detection rate") + 
    ylab("Expression (log2)") + 
    labs(title = paste0(title.deng, " (Group B)")) + 
    theme(plot.title = element_text(size =20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank(), 
          legend.position = "none")

dat <- tidyQuants(eDeng[, pdDeng$bin == "Bin C"], 
                  CDRval = pdDeng$CDRlt1[pdDeng$bin == "Bin C"], 
                  qRange = qVals)
pMed.Deng.C <- ggplot(dat, aes(x = CDR, y = value, colour = Quantile)) +
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    geom_point(size = 2) + 
    xlab("Detection rate") + 
    ylab("Expression (log2)") + 
    labs(title = paste0(title.deng, " (Group C)")) + 
    theme(plot.title = element_text(size =20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank(), 
          legend.position = "none")

dat <- tidyQuants(eDeng[, pdDeng$bin == "Bin D"], 
                  CDRval = pdDeng$CDRlt1[pdDeng$bin == "Bin D"], 
                  qRange = qVals)
pMed.Deng.D <- ggplot(dat, aes(x = CDR, y = value, colour = Quantile)) +
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    geom_point(size = 2) + 
    xlab("Detection rate") + 
    ylab("Expression (log2)") + 
    labs(title = paste0(title.deng, " (Group D)")) + 
    theme(plot.title = element_text(size =20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank(),  
          legend.position = "none")

dat <- tidyQuants(eGuo[, pdGuo$weekGroup == "4"], 
                  CDRval = pdGuo$CDRlt1[pdGuo$weekGroup == "4"], 
                  qRange = qVals)
pMed.Guo.A <- ggplot(dat, aes(x = CDR, y = value, colour = Quantile)) +
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.8, size = 2) + 
    geom_point(size = 2) + 
    xlab("Detection rate") + 
    ylab("Expression (log2)") + 
    labs(title = paste0(title.guo, " (Group A)")) + 
    theme(plot.title = element_text(size =20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank(), 
          legend.position = "none")

dat <- tidyQuants(eGuo[, pdGuo$weekGroup == "7-8"], 
                  CDRval = pdGuo$CDRlt1[pdGuo$weekGroup == "7-8"], 
                  qRange = qVals)
pMed.Guo.B <- ggplot(dat, aes(x = CDR, y = value, colour = Quantile)) +
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    geom_point(size = 2) + 
    xlab("Detection rate") + 
    ylab("Expression (log2)") + 
    labs(title = paste0(title.guo, " (Group B)")) + 
    theme(plot.title = element_text(size =20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank(),  
          legend.position = "none")

dat <- tidyQuants(eGuo[, pdGuo$weekGroup == "10-11"], 
                  CDRval = pdGuo$CDRlt1[pdGuo$weekGroup == "10-11"], 
                  qRange = qVals)
pMed.Guo.C <- ggplot(dat, aes(x = CDR, y = value, colour = Quantile)) +
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    geom_point(size = 2) + 
    xlab("Detection rate") + 
    ylab("Expression (log2)") + 
    labs(title = paste0(title.guo, " (Group C)")) + 
    theme(plot.title = element_text(size =20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank(), 
          legend.position = "none")

dat <- tidyQuants(eKowalczyk, CDRval = pdKowalczyk$CDRlt1, qRange = qVals)
pMed.Kowalczyk <- ggplot(dat, aes(x = CDR, y = value, colour = Quantile)) +
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    geom_point(size = 2) + 
    xlab("Detection rate") + 
    ylab("Expression (log2)") + 
    labs(title = title.kowalczyk) + 
    theme(plot.title = element_text(size =20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank(), 
          legend.position = "none")

dat <- tidyQuants(eKumar[, pdKumar$bin == "Group A"], 
                  CDRval = pdKumar$CDRlt1[pdKumar$bin == "Group A"], qRange = qVals)
pMed.Kumar.A <- ggplot(dat, aes(x = CDR, y = value, colour = Quantile)) +
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    geom_point(size = 2) + 
    xlab("Detection rate") + 
    ylab("Expression (log2)") + 
    labs(title = paste0(title.kumar, " (Group A)")) + 
    theme(plot.title = element_text(size =20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank(),  
          legend.position = "none")

dat <- tidyQuants(eKumar[, pdKumar$bin == "Group B"], 
                  CDRval = pdKumar$CDRlt1[pdKumar$bin == "Group B"], qRange = qVals)
pMed.Kumar.B <- ggplot(dat, aes(x = CDR, y = value, colour = Quantile)) +
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    geom_point(size = 2) + 
    xlab("Detection rate") + 
    ylab("Expression (log2)") + 
    labs(title = paste0(title.kumar, " (Group B)")) + 
    theme(plot.title = element_text(size =20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank(), 
          legend.position = "none")

dat <- tidyQuants(eKumar[, pdKumar$bin == "Group C"], 
                  CDRval = pdKumar$CDRlt1[pdKumar$bin == "Group C"], qRange = qVals)
pMed.Kumar.C <- ggplot(dat, aes(x = CDR, y = value, colour = Quantile)) +
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    geom_point(size = 2) + 
    xlab("Detection rate") + 
    ylab("Expression (log2)") + 
    labs(title = paste0(title.kumar, " (Group A)")) + 
    theme(plot.title = element_text(size =20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank(), 
          legend.position = "none")



ID <- patel_pd$includeSample == TRUE & 
  patel_pd$sampleType == "SC" & patel_pd$cellType == "Glioblastoma"
esetSub <- as.matrix(patel_tpm[, ID])
esetSub <- log2(esetSub + 1)
pdSub <- patel_pd[ID, ]

dat <- tidyQuants(esetSub, CDRval = pdSub$CDRlt1, qRange = qVals)
pMed.Patel <- ggplot(dat, aes(x = CDR, y = value, colour = Quantile)) +
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    geom_point(size = 2) + 
    xlab("Detection rate") + 
    ylab("Expression (log2)") + 
    labs(title = title.patel) + 
    theme(plot.title = element_text(size =20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank(), 
          legend.position = "none")

dat <- tidyQuants(eTreutlein, CDRval = pdTreutlein$CDRlt1, qRange = qVals)
pMed.Treutlein <- ggplot(dat, aes(x = CDR, y = value, colour = Quantile)) +
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    geom_point(size = 2) + 
    xlab("Detection rate") + 
    ylab("Expression (log2)") + 
    labs(title = title.treutlein) + 
    theme(plot.title = element_text(size =20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank(),  
          legend.position = "none")

dat <- tidyQuants(eShalek, CDRval = pdShalek$CDRlt1, qRange = qVals)
pMed.Shalek <- ggplot(dat, aes(x = CDR, y = value, colour = Quantile)) +
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    geom_point(size = 2) + 
    xlab("Detection rate") + 
    ylab("Expression (log2)") + 
    labs(title = title.shalek.2014) + 
    theme(plot.title = element_text(size =20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank(), 
          legend.position = "none")

dat <- tidyQuants(eTrapnell, CDRval = pdTrapnell$CDRlt1, qRange = qVals)
pMed.Trapnell <- ggplot(dat, aes(x = CDR, y = value, colour = Quantile)) +
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    geom_point(size = 2) + 
    xlab("Detection rate") + 
    ylab("Expression (log2)") + 
    labs(title = title.trapnell) + 
    theme(plot.title = element_text(size =20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank(), 
          legend.position = "none")


dat <- tidyQuants(eBurns[, pdBurns$tissue == "cochlear"], 
                  CDRval = pdBurns$CDRlt1[pdBurns$tissue == "cochlear"], qRange = qVals)
pMed.Burns.A <- ggplot(dat, aes(x = CDR, y = value, colour = Quantile)) +
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    geom_point(size = 2) + 
    xlab("Detection rate") + 
    ylab("Expression (log2)") + 
    labs(title = paste0(title.burns, " (Group A)")) + 
    theme(plot.title = element_text(size =20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank(), 
          legend.position = "none")

dat <- tidyQuants(eBurns[, pdBurns$group == "utricular_HC"], 
                  CDRval = pdBurns$CDRlt1[pdBurns$group == "utricular_HC"], qRange = qVals)
pMed.Burns.B <- ggplot(dat, aes(x = CDR, y = value, colour = Quantile)) +
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    geom_point(size = 2) + 
    xlab("Detection rate") + 
    ylab("Expression (log2)") + 
    labs(title = paste0(title.burns, " (Group B)")) + 
    theme(plot.title = element_text(size =20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank(), 
          legend.position = "none")

dat <- tidyQuants(eBurns[, pdBurns$group == "utricular_SC"], 
                  CDRval = pdBurns$CDRlt1[pdBurns$group == "utricular_SC"], qRange = qVals)
pMed.Burns.C <- ggplot(dat, aes(x = CDR, y = value, colour = Quantile)) +
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    geom_point(size = 2) + 
    xlab("Detection rate") + 
    ylab("Expression (log2)") + 
    labs(title = paste0(title.burns, " (Group C)")) + 
    theme(plot.title = element_text(size =20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank(), 
          legend.position = "none")

dat <- tidyQuants(eBurns[, pdBurns$group == "utricular_TEC"], 
                  CDRval = pdBurns$CDRlt1[pdBurns$group == "utricular_TEC"], qRange = qVals)
pMed.Burns.D <- ggplot(dat, aes(x = CDR, y = value, colour = Quantile)) +
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    geom_point(size = 2) + 
    xlab("Detection rate") + 
    ylab("Expression (log2)") + 
    labs(title = paste0(title.burns, " (Group D)")) + 
    theme(plot.title = element_text(size =20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank(), 
          legend.position = "none")

dat <- tidyQuants(eLeng, CDRval = pdLeng$CDRlt1, qRange = qVals)
pMed.Leng <- ggplot(dat, aes(x = CDR, y = value, colour = Quantile)) +
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    geom_point(size = 2) + 
    xlab("Detection rate") + 
    ylab("Expression (log2)") + 
    labs(title = title.leng) + 
    theme(plot.title = element_text(size =20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank(), 
          legend.position = "none")


dat <- tidyQuants(eBose, CDRval = pdBose$CDRlt1, qRange = qVals)
pMed.Bose <- ggplot(dat, aes(x = CDR, y = value, colour = Quantile)) +
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    geom_point(size = 2) + 
    xlab("Detection rate") + ylab("Expression (log2)") + 
    labs(title = title.bose) + 
    theme(plot.title = element_text(size =20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank(),  
          legend.position = "none")


dat <- tidyQuants(eJaitin, CDRval = pdJaitin$CDRlt1, qRange = qVals)
pMed.Jaitin <- ggplot(dat, aes(x = CDR, y = value, colour = Quantile)) +
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    geom_point(size = 2) + 
    xlab("Detection rate") + ylab("Expression (log2)") + 
    labs(title = title.jaitin) + 
    theme(plot.title = element_text(size =20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank(),  
          legend.position = "none")

dat <- tidyQuants(eMacosko, CDRval = pdMacosko$CDRlt1, qRange = qVals)
pMed.Macosko <- ggplot(dat, aes(x = CDR, y = value, colour = Quantile)) +
     stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.2, size = 2) + 
    geom_point(size = 2) + 
    xlab("Detection rate") + ylab("Expression (log2)") + 
    labs(title = title.macosko) + 
    theme(plot.title = element_text(size =20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank(),  
          legend.position = "none")


dat <- tidyQuants(eSatija, CDRval = pdSatija$CDRlt1, qRange = qVals)
pMed.Satija <- ggplot(dat, aes(x = CDR, y = value, colour = Quantile)) +
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    geom_point(size = 2) + 
    xlab("Detection rate") + ylab("Expression (log2)") + 
    labs(title = title.satija) + 
    theme(plot.title = element_text(size =20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank(),  
          legend.position = "none")

dat <- tidyQuants(eZeisel[, pdZeisel$tissue == "sscortex"], 
                  CDRval = pdZeisel$CDRlt1[pdZeisel$tissue == "sscortex"], 
                  qRange = qVals)
pMed.Zeisel.A <- ggplot(dat, aes(x = CDR, y = value, colour = Quantile)) +
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    geom_point(size = 2) + 
    xlab("Detection rate") + ylab("Expression (log2)") + 
    labs(title = "Zeisel et al., 2015 (Group A)") + 
    theme(plot.title = element_text(size =20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank(), 
          legend.position = "none")

dat <- tidyQuants(eZeisel[, pdZeisel$tissue == "ca1hippocampus"], 
                  CDRval = pdZeisel$CDRlt1[pdZeisel$tissue == "ca1hippocampus"], 
                  qRange = qVals)
pMed.Zeisel.B <- ggplot(dat, aes(x = CDR, y = value, colour = Quantile)) +
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    geom_point(size = 2) + 
    xlab("Detection rate") + ylab("Expression (log2)") + 
    labs(title = "Zeisel et al., 2015 (Group B)") + 
    theme(plot.title = element_text(size =20),
          axis.text = element_text(size = 20), 
          axis.title = element_text(size =20), 
          legend.text = element_text(size =40), 
          legend.title = element_text(size =40), 
          legend.position = "top")
legend <- get_legend(pMed.Zeisel.B)

pMed.Zeisel.B <- ggplot(dat, aes(x = CDR, y = value, colour = Quantile)) +
    stat_smooth(method = "loess", formula = y ~ poly(x, degree = 1), 
                se = FALSE, span = 0.75, size = 2) + 
    geom_point(size = 2) + 
    xlab("Detection rate") + ylab("Expression (log2)") + 
    labs(title = "Zeisel et al., 2015 (Group B)") + 
    theme(plot.title = element_text(size =20),
          axis.text = element_text(size = 20), 
          axis.title = element_blank(), 
          legend.position = "none")

pout <- plot_grid(pMed.Deng.D, pMed.Guo.A , pMed.Kowalczyk, pMed.Kumar.C, pMed.Patel,
                        pMed.Treutlein, pMed.Shalek, pMed.Trapnell, pMed.Burns.A, pMed.Leng,
                        pMed.Bose, pMed.Jaitin, pMed.Macosko, pMed.Satija, pMed.Zeisel.B, 
                        labels = LETTERS[1:15], label_size = 25, ncol = 5)


p <- plot_grid(legend, pout, ncol = 1, rel_heights = c(.5, 4))
pdf(file.path(working_path, "scBatchSuppFig-kaboxplot.pdf"), width = 30, height = 18)
print(p)
dev.off()

```




# Figure: Discovered cell types from Burns et al. (2015) 
```{r}

ID = pdBurns$tissue %in% "cochlear" & !(pdBurns$celltypespecific %in% c("HC", "NSC (i)", "NSC (ii)", "SC", "Med SC"))
esetSub <- eBurns[, ID]
esetSub <- sweep(esetSub, 2, colMeans(esetSub), FUN = "-")
pdSub <- pdBurns[ID,]
pdSub$fluidicChip <- factor(pdSub$fluidicChip)
pdSub$fluidicChip.sort = factor(pdSub$fluidicChip, 
                levels = names(sort(unlist(lapply( split(pdSub$CDRlt1, pdSub$fluidicChip), median)), decreasing = FALSE)))
pdSub$celltypespecific.sort = factor(pdSub$celltypespecific, 
                levels = names(sort(unlist(lapply( split(pdSub$CDRlt1, pdSub$celltypespecific), median)), decreasing = FALSE)))

# fisher.test(pdSub$celltypespecific.sort, pdSub$fluidicChip.sort)

s <- svd(esetSub - rowMeans(esetSub))
pcVar <- c(s$d^2 / sum(s$d^2))[1:3]
scores <- data.frame(pdSub, s$v[,1:3])
if(cor(scores$X1, scores$CDRlt1) < 0){ scores$X1 <- scores$X1*(-1) } 


p3 <- scores %>% ggplot(aes(x = celltypespecific.sort, y=CDRlt1)) + 
        geom_boxplot(outlier.shape = NA) + 
        geom_jitter(size = 2, width = 0.2) + 
        ylab("Detection rate") + xlab("Discovered Celltypes") +
        # labs(title = "Burns et al. (2015) - Cochlear SCs")  + 
        scale_color_discrete(guide = guide_legend(title = "Batch")) + 
        theme(plot.title = element_text(size = 20), axis.text = element_text(size = 20),
          axis.title = element_text(size = 20), legend.text = element_text(size = 15), 
          legend.title = element_text(size = 15)) 
  
p4 <- scores %>% ggplot(aes(x = fluidicChip.sort, y=CDRlt1)) + 
        geom_boxplot(outlier.shape = NA) + 
        geom_jitter(size = 2, width = 0.2) + 
        ylab("Detection rate") + xlab("Batch") + # theme(axis.text.x = element_text(angle = 45)) + 
        # labs(title = "Burns et al. (2015) - Cochlear SCs")  + 
        scale_color_discrete(guide = guide_legend(title = "Discovered\nCelltypes")) + 
        theme(plot.title = element_text(size = 20), axis.text = element_text(size = 20),
          axis.title = element_text(size = 20), legend.text = element_text(size = 15), 
          legend.title = element_text(size = 15))

p.4.1 <- scores %>% 
  group_by(celltypespecific.sort,fluidicChip.sort) %>% 
  summarise(count=n()) %>% 
  dplyr::mutate(perc=count/sum(count)) %>% 
  ggplot(aes(x =factor(celltypespecific.sort), y = perc*100, fill = factor(fluidicChip.sort))) +
  geom_bar(stat="identity") +
  labs(x = "Discovered Celltypes", y = "Percentage", fill = "Batch") + 
  # scale_fill_discrete(guide = guide_legend(title = "Batch")) + 
  theme(plot.title = element_text(size = 20), axis.text = element_text(size = 20),
          axis.title = element_text(size = 20), legend.text = element_text(size = 15), 
          legend.title = element_text(size = 15)) 


# p.pc <- ggplot(scores, aes(x = X1, y = X2, color = fluidicChip.sort)) + 
#     geom_point(size = 4) + 
#     xlab(paste0("Principal Component 1 (", round(pcVar[1],2)*100, "%)")) + 
#     ylab(paste0("Principal Component 2 (", round(pcVar[2],2)*100, "%)")) + 
#     scale_color_discrete(guide = guide_legend(title = "Batch"))

p.CDRlt1vspc <- ggplot(scores, aes(x = CDRlt1, y = X1, 
              shape = celltypespecific.sort, color = fluidicChip.sort)) + 
    geom_point(size = 5) + xlab("Detection rate") + 
  ylab(paste0("Principal Component 1 (", round(pcVar[1],2)*100, "%)"))  + 
  theme(plot.title = element_text(size = 20), axis.text = element_text(size = 20),
          axis.title = element_text(size = 20), legend.text = element_text(size = 15), 
          legend.title = element_text(size = 15)) + 
  scale_color_discrete(guide = guide_legend(title = "Batch")) + 
  scale_shape_discrete(guide = guide_legend(title = "Celltype"))


p.out <- plot_grid(p3, p4, p.4.1, p.CDRlt1vspc, ncol = 2,
                        labels = LETTERS[1:4], label_size = 25)


pdf(file.path(working_path, "scBatchSuppFig-Burns-discover.pdf"), width = 15, height = 13)
print(p.out)
dev.off()

```


# Figure: Discovered cell types from Zeisel et al. (2014)
```{r Supp Figure box zeisel, fig.width = 12, fig.height = 10}
ff = ffprint <- levels(factor(pdZeisel$level1class))
ffprint[6:7] <- c("pyramidalCA1", "pyramidalSS")
i = 7
pdSub <- pdZeisel[pdZeisel$level1class == ff[i] & !(pdZeisel$level2class == "(none)"),] 
pdSub$runID.sort = factor(pdSub$runID, levels = names(sort(unlist(lapply( split(pdSub$CDRlt1, pdSub$runID), median)), 
                                      decreasing = FALSE)))
pdSub$level2class.sort = factor(pdSub$level2class, 
                                levels = names(sort(unlist(lapply( split(pdSub$CDRlt1, pdSub$level2class), median)), 
                                                    decreasing = FALSE)))
esetSub <- eZeisel[, pdZeisel$level1class == ff[i] & !(pdZeisel$level2class == "(none)")]
esetSub <- sweep(esetSub, 2, colMeans(esetSub), FUN = "-")
s <- svd(esetSub - rowMeans(esetSub))
pcVar <- c(s$d^2 / sum(s$d^2))[1:3]
scores <- data.frame(pdSub, s$v[,1:3])
if(cor(scores$X1, scores$CDRlt1) < 0){ scores$X1 <- scores$X1*(-1) } 
p1 <- pdSub %>% ggplot(aes(x = level2class.sort, y=CDRlt1)) + 
     geom_boxplot(outlier.shape = NA) + 
     geom_jitter(width = 0.2) + 
     ylab("Detection rate") + xlab("Discovered Celltypes") + # scale_y_log10() + 
     scale_color_discrete(guide = guide_legend(title = "Batch")) + 
     # labs(title = paste0("Zeisel et al. (2015) - ", ffprint[i]))  + 
     theme(plot.title = element_text(size = 20), axis.text = element_text(size = 20),
       axis.title = element_text(size = 20), legend.text = element_text(size = 15), 
       legend.title = element_text(size = 15), 
       legend.position = "none") 


p2 <- pdSub %>% ggplot(aes(x = runID.sort, y=CDRlt1)) + 
    geom_boxplot(outlier.shape = NA) + 
    geom_jitter(width = 0.2) + 
    ylab("Detection rate") + xlab("Batch") + # theme(axis.text.x = element_text(angle = 45)) + 
       # scale_color_discrete(guide = guide_legend(title = "Discovered\nCelltypes")) + # scale_y_log10() + 
       theme(plot.title = element_text(size = 20), axis.text = element_text(size = 20),
         axis.title = element_text(size = 20), legend.text = element_text(size = 15), 
         legend.title = element_text(size = 15), 
         legend.position = "none")
   
p3 <- pdSub %>% 
         group_by(level2class.sort, runID.sort) %>% 
         summarise(count=n()) %>% 
         dplyr::mutate(perc=count/sum(count)) %>% 
         ggplot(aes(x =factor(level2class.sort), y = perc*100, fill = factor(runID.sort))) +
         geom_bar(stat="identity") +
         labs(x = "Discovered Celltypes", y = "Percentage", fill = "Batch") + 
    theme(plot.title = element_text(size = 20), axis.text = element_text(size = 20),
       axis.title = element_text(size = 20), legend.text = element_text(size = 15), 
       legend.title = element_text(size = 15))

 
p.box <- plot_grid(p2, p1, p3, ncol = 1, labels = LETTERS[1:3], label_size = 25)

pdf(paste0(file.path(working_path, "scBatchSuppFig-Zeisel-boxPercent-"), 
           ffprint[i],".pdf"), width = 16, height = 18)
print(p.box)
dev.off()

```



# Figure - Boxplots from Deng et al. (2014)
```{r Supp Figure box deng, fig.width=18, fig.height=22, warning=FALSE}
scores <- data.frame(pdDeng)
scores$runID = factor(scores$runID, levels = names(sort(unlist(lapply( split(scores$CDRlt1, scores$runID), median)), 
                                          decreasing = FALSE)))
box.A.left <- scores %>% dplyr::filter(bin == "Bin A") %>% ggplot(aes(x = time, y = CDRlt1, fill = runID)) + 
    geom_boxplot() + xlab(" ") + ylab(" ") + 
    labs(title = "Deng et al. (2014) - Group A")  + 
    ylab("Detection rate") + 
        theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20),
          axis.title = element_text(size = 20), 
          legend.text = element_text(size = 15), 
          legend.title = element_text(size = 15)) + 
    geom_vline(xintercept=c(1.5))

box.A.right <- scores %>% dplyr::filter(bin == "Bin A") %>% 
    ggplot(aes(x = runID, y = CDRlt1, fill = time)) + 
    geom_boxplot() + xlab(" ") + ylab(" ") + 
    labs(title = "Deng et al. (2014) - Group A")  + 
    ylab("Detection rate") + 
        theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20),
          axis.title = element_text(size = 20), 
          legend.text = element_text(size = 15), 
          legend.title = element_text(size = 15)) + 
    geom_vline(xintercept=c(1.5))

sub1 <- scores %>% dplyr::filter(bin == "Bin B") 
sub <- scores %>% dplyr::filter(bin == "Bin B", time == "Late 2-cell", runID != "Run0084") 
cols = gg_color_hue(4)
box.B.left <- ggplot(sub1, aes(x = time, y = CDRlt1)) + 
    geom_boxplot(aes(fill = runID)) + xlab(" ") + ylab(" ") + 
    labs(title = "Deng et al. (2014) - Group B")  + 
    ylab("Detection rate") + 
        theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20),
          axis.title = element_text(size = 20), 
          legend.text = element_text(size = 15), 
          legend.title = element_text(size = 15)) + 
    geom_vline(xintercept=c(1.5)) + 
    geom_point(aes(x = 2.25, y = sub[2,]$CDRlt1), color = cols[2], size = 5) + 
    geom_point(aes(x = 1.75, y = sub[1,]$CDRlt1), color = cols[4], size = 5) 

sub1 <- scores %>% dplyr::filter(bin == "Bin B") 
sub <- scores %>% dplyr::filter(bin == "Bin B", time == "Late 2-cell", runID %in% c("Run0083", "Run0085"))
cols = gg_color_hue(2)
box.B.right <- ggplot(sub1, aes(x = runID, y = CDRlt1)) + 
    geom_boxplot(aes(fill = time)) + xlab(" ") + ylab(" ") + 
    labs(title = "Deng et al. (2014) - Group B")  + 
    ylab("Detection rate") + 
        theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20),
          axis.title = element_text(size = 20), 
          legend.text = element_text(size = 15), 
          legend.title = element_text(size = 15)) + 
    geom_vline(xintercept=c(1.5, 2.5, 3.5)) + 
    geom_point(aes(x = 4.2, y = sub[2,]$CDRlt1), color = cols[2], size = 5) + 
    geom_point(aes(x = 2.0, y = sub[1,]$CDRlt1), color = cols[2], size = 5) 


box.C.left <- scores %>% dplyr::filter(bin == "Bin C") %>% 
    ggplot(aes(x = time, y = CDRlt1, fill = runID)) + 
    geom_boxplot() + xlab(" ") + ylab(" ") + 
    labs(title = "Deng et al. (2014) - Group C")  + 
    ylab("Detection rate") +  
        theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20),
          axis.title = element_text(size = 20),
          legend.text = element_text(size = 15), 
          legend.title = element_text(size = 15)) + ylim(0.30, 0.45) + 
    geom_vline(xintercept=c(1.5, 2.5))


box.C.right <- scores %>% dplyr::filter(bin == "Bin C") %>% 
    ggplot(aes(x = runID, y = CDRlt1, fill = time)) + 
    geom_boxplot() + xlab(" ") + ylab(" ") + 
    labs(title = "Deng et al. (2014) - Group C")  + 
    ylab("Detection rate") + 
        theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20),
          axis.title = element_text(size = 20), 
          axis.text.x = element_text(angle = 45, hjust = 1, size = 10),
          legend.text = element_text(size = 15),
          legend.title = element_text(size = 15)) + ylim(0.30, 0.45) + 
    geom_vline(xintercept=c(1.5, 2.5, 3.5, 4.5, 5.5, 6.5))


box.D.left <- scores %>% dplyr::filter(bin == "Bin D") %>% 
    ggplot(aes(x = time, y = CDRlt1, fill = runID)) + 
    geom_boxplot() + xlab(" ") + ylab(" ") + 
    labs(title = "Deng et al. (2014) - Group D")  + 
    ylab("Detection rate") + 
        theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20),
          axis.title = element_text(size = 20),
          legend.text = element_text(size = 15), 
          legend.title = element_text(size = 15)) + ylim(0.25, 0.45) + 
    geom_vline(xintercept=c(1.5, 2.5))


box.D.right <- scores %>% dplyr::filter(bin == "Bin D") %>% 
    ggplot(aes(x = runID, y = CDRlt1, fill = time)) + 
    geom_boxplot() + xlab(" ") + ylab(" ") + 
    labs(title = "Deng et al. (2014) - Group D")  + 
    ylab("Detection rate") + 
        theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20),
          axis.title = element_text(size = 20), 
          axis.text.x = element_text(size = 15),
          legend.text = element_text(size = 15), 
          legend.title = element_text(size = 15)) + ylim(0.25, 0.45) + 
    geom_vline(xintercept=c(1.5, 2.5, 3.5))


p1 <- plot_grid(box.A.left, box.A.right, 
                box.B.left, box.B.right,
                box.C.left, box.C.right, 
                box.D.left, box.D.right, ncol = 2, 
                labels =LETTERS[1:8], label_size = 20)
p1 

pdf(file.path(working_path, "scBatchSuppFig-Deng-box.pdf"), width = 18, height = 22)
print(p1)
dev.off()

```

# Figure - Boxplots from Guo et al. (2015)
```{r Supp Figure box guo, fig.width = 15, fig.height = 18}
levels(pdGuo$batch) <- paste("Batch", 1:14)
scores <- data.frame(pdGuo)
scores$batch <- factor(scores$batch)
scores$batch = factor(scores$batch, levels = names(sort(unlist(lapply( split(scores$CDRlt1, scores$batch), median)), 
                                          decreasing = FALSE)))

box.A.left <- scores %>% dplyr::filter(weekGroup == "4") %>% ggplot(aes(x = week, y = CDRlt1, fill = batch)) + 
    geom_boxplot() + xlab(" ") + ylab(" ") + 
    labs(title = "Guo et al. (2015) - Week 4")  + 
    ylab("Detection rate") + ylim(c(0.2, 0.43)) +  
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20),
          axis.title = element_text(size = 20), 
          legend.text = element_text(size = 15), 
          legend.title = element_text(size = 15)) 

box.A.right <- scores %>% dplyr::filter(weekGroup == "4") %>% 
    ggplot(aes(x = batch, y = CDRlt1, fill = week)) + 
    geom_boxplot() + xlab(" ") + ylab(" ") + 
    labs(title = "Guo et al. (2015) - Week 4")  + 
    ylab("Detection rate") + ylim(c(0.2, 0.43)) +  
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20),
          axis.title = element_text(size = 20), 
          legend.text = element_text(size = 15), 
          legend.title = element_text(size = 15)) + 
    geom_vline(xintercept=seq(1.5, 2.5, by=1))

cols = gg_color_hue(8)
subscores <- scores %>% dplyr::filter(weekGroup == "7-8") 
subscores$batch = factor(subscores$batch, levels = names(sort(unlist(lapply( split(subscores$CDRlt1, subscores$batch), median)), 
                                          decreasing = FALSE)))
sub <- scores %>% dplyr::filter(week == "8", batch == "Batch 5") 
box.B.left <- subscores %>%
    ggplot(aes(x = week, y = CDRlt1, fill = batch)) + 
    geom_boxplot() + xlab(" ") + ylab(" ") + 
    labs(title = "Guo et al. (2015) - Weeks 7-8")  + 
    ylab("Detection rate") + ylim(c(0.2, 0.43)) + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20),
          axis.title = element_text(size = 20), 
          legend.text = element_text(size = 15), 
          legend.title = element_text(size = 15)) + 
    geom_vline(xintercept=c(1.5)) + 
    geom_point(aes(x = 2.10, y = sub[1,]$CDRlt1), color = cols[7], size = 5, show.legend = FALSE)

cols = gg_color_hue(2)
box.B.right <- subscores %>%
    ggplot(aes(x = batch, y = CDRlt1, fill = week)) + 
    geom_boxplot() + xlab(" ") + ylab(" ") + 
    labs(title = "Guo et al. (2015) - Weeks 7-8")  + 
    ylab("Detection rate") + ylim(c(0.2, 0.43)) +  
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20),
          axis.title = element_text(size = 20), 
          axis.text.x = element_text(angle = 45, hjust = 1, size = 15),
          legend.text = element_text(size = 15), 
          legend.title = element_text(size = 15)) + 
    geom_vline(xintercept=seq(1.5, 7.5, by=1)) + 
    geom_point(aes(x = 7, y = sub[1,]$CDRlt1), color = cols[2], size = 5, show.legend = FALSE)

cols = gg_color_hue(7)
subscores <- scores %>% dplyr::filter(weekGroup == "10-11") 
subscores$batch = factor(subscores$batch, levels = names(sort(unlist(lapply( split(subscores$CDRlt1, subscores$batch), median)), 
                                          decreasing = FALSE)))

subscores$batch <- factor(subscores$batch, levels(subscores$batch)[c(1:5, 7, 6)])
sub1 <- scores %>% dplyr::filter(week == "10", batch == "Batch 11") 
sub2 <- scores %>% dplyr::filter(week == "10", batch == "Batch 12") 
box.C.left <- subscores %>%
    ggplot(aes(x = week, y = CDRlt1, fill = batch)) + 
    geom_boxplot() + xlab(" ") + ylab(" ") + 
    labs(title = "Guo et al. (2015) - Weeks 10-11")  + 
    ylab("Detection rate") + ylim(c(0.2, 0.43)) +  
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20),
          axis.title = element_text(size = 20), 
          legend.text = element_text(size = 15), 
          legend.title = element_text(size = 15)) + 
    geom_vline(xintercept=c(1.5)) + 
    geom_point(aes(x = 1.3, y = sub1[1,]$CDRlt1), color = cols[7], size = 5, show.legend = FALSE) + 
    geom_point(aes(x = 1.15, y = sub2[1,]$CDRlt1), color = cols[6], size = 5, show.legend = FALSE)
    

cols = gg_color_hue(2)
box.C.right <- subscores %>%
    ggplot(aes(x = batch, y = CDRlt1, fill = week)) + 
    geom_boxplot() + xlab(" ") + ylab(" ") + 
    labs(title = "Guo et al. (2015) - Weeks 10-11")  + 
    ylab("Detection rate") + ylim(c(0.2, 0.43)) +  
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20),
          axis.title = element_text(size = 20), 
          axis.text.x = element_text(angle = 45, hjust = 1, size = 15),
          legend.text = element_text(size = 15), 
          legend.title = element_text(size = 15)) + 
    geom_vline(xintercept=seq(1.5, 6.5, by=1)) + 
    geom_point(aes(x = 7, y = sub1[1,]$CDRlt1), color = cols[1], size = 5, show.legend = FALSE) + 
    geom_point(aes(x = 5.8, y = sub2[1,]$CDRlt1), color = cols[1], size = 5, show.legend = FALSE)


p1 <- plot_grid(box.A.left, box.A.right, 
                box.B.left, box.B.right,
                box.C.left, box.C.right, ncol = 2, 
                labels =LETTERS[1:6], label_size = 20)
p1 

pdf(file.path(working_path, "scBatchSuppFig-Guo-box.pdf"), width = 15, height = 18)
print(p1)
dev.off()

```

# Figure - Boxplots from Kowalczyk et al. (2015)
```{r Supp Figure box kowalczyk, fig.width = 12, fig.height = 10}
scores <- data.frame(pdKowalczyk)
scores$batch <- factor(scores$batch)
levels(scores$batch) <- paste("Batch", 1:7)
scores$batch = factor(scores$batch, levels = names(sort(unlist(lapply( split(scores$CDRlt1, scores$batch), median)), 
                                          decreasing = FALSE)))
scores$age <- factor(scores$age, levels = c("2-3 months", "22 months "))

cols = gg_color_hue(7)
sub <- scores %>% dplyr::filter(age == "22 months ", batch == "Batch 5") 
p.box.batch <- ggplot(scores, aes(age, CDRlt1, fill = batch)) + geom_boxplot() + 
    theme(axis.text.x = element_text(angle = 90, hjust = 1)) + 
    labs(title = "Kowalczyk et al. (2015)") +
    ylab("Detection Rate") + 
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_text(size = 20), 
          legend.text = element_text(size = 15), 
          legend.title = element_text(size = 15)) + 
    geom_vline(xintercept=c(1.5)) + 
    geom_point(aes(x = 2.15, y = sub[1,]$CDRlt1), color = cols[6], size = 5, show.legend = FALSE) 
p.box.batch

pdf(file.path(working_path, "scBatchSuppFig-Kowalczyk-box.pdf"), width = 12, height = 10)
print(p.box.batch)
dev.off()

```

# Figure - Boxplots from Kumar et al. (2014)
```{r Supp Figure box kumar, fig.width = 12, fig.height = 10}
scores <- data.frame(pdKumar)

box.left <- scores %>% 
    ggplot(aes(x = bin, y = CDRlt1, fill = batch)) + 
    geom_boxplot() + xlab(" ") + ylab(" ") + 
    labs(title = "Kumar et al. (2014)") + 
    ylab("Detection Rate") +  
        theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20),
          axis.title = element_text(size = 20), 
          legend.text = element_text(size = 15), 
          legend.title = element_text(size = 15)) + 
    geom_vline(xintercept=seq(1.5, 2.5, by=1))

box.left

pdf(file.path(working_path, "scBatchSuppFig-Kumar-box.pdf"), width = 12, height = 10)
print(box.left)
dev.off()

```

# Figure - Boxplots from Patel et al. (2014)
```{r Supp Figure Patel boxplot, fig.width=12, fig.height=10}
scores <- data.frame(pdPatel)
p.boxplot <- ggplot(pdPatel, aes(tumorName, CDRlt1, fill = runID)) + geom_boxplot() +
    xlab("Tumor") + ylab("Detection rate") +
    labs(title = "Patel et al. (2014)") + 
    geom_vline(xintercept=seq(1.5, 4.5, by=1))
p.boxplot

pdf(file.path(working_path, "scBatchSuppFig-Patel-box.pdf"), width = 12, height = 10)
print(p.boxplot)
dev.off()

```

# Figure - Boxplots from Treutlein et al. (2014)
```{r Supp Figure box treutlein, fig.width = 12, fig.height = 10}
scores <- data.frame(pdTreutlein)
scores$batch = factor(scores$batch, levels = names(sort(unlist(lapply( split(scores$CDRlt1, scores$batch), median)), 
                                          decreasing = FALSE)))
cols = gg_color_hue(7)
sub <- scores %>% dplyr::filter(day == "ED14.5", batch == "Batch 1") 
p.box.batch <- ggplot(scores, aes(x = day, y = CDRlt1, fill = batch)) + 
    geom_boxplot() + xlab(" ") + ylab(" ") + 
    labs(title = "Treutlein et al. (2014)") +
    ylab("Detection rate") +
    theme(plot.title = element_text(size = 20),
          axis.text = element_text(size = 20), 
          axis.title = element_text(size = 20), 
          legend.text = element_text(size = 15), 
          legend.title = element_text(size = 15)) + 
    geom_vline(xintercept=seq(1.5, 3.5, by=1)) + 
    geom_point(aes(x = 0.80, y = sub[1,]$CDRlt1), color = cols[1], size = 5, show.legend = FALSE) 
p.box.batch

pdf(file.path(working_path, "scBatchSuppFig-Treutlein-box.pdf"), width = 12, height = 10)
print(p.box.batch)
dev.off()

```

# Figure - Bulk vs SC scatter plot
```{r}
pdf(file.path(working_path, "Figure_scatter-withblueline.pdf"), width = 18, height =6) 
par(mfrow=c(1,3), mai=c(0.75,0.75,0.4,0.1), mgp=c(3.3,1.25,0),
    cex.lab=2.25, cex.axis=2.25, cex.main=2.25, cex.sub=2.25)

#### Shalek et al. (2013)
library(scRNASeqMouseShalekBMDCs)
data("BMDCsMouseTPM")

eset <- exprs(BMDCsMouseTPM)
pd <- pData(BMDCsMouseTPM)

esetBulk <- eset[, pd$sampleType == "bulk"]
esetSC <- eset[, pd$sampleType == "SC"]

xx = esetBulk[,1];
yy = esetSC
smDat <- sm(x=xx, y =rowMeans(yy), n.bins = 50)
plot(x = xx, y = rowMeans(yy), col = "gray", pch = ".", 
     xlim = c(0, 30), ylim = c(0, 30),  
     xlab = "Bulk profile", ylab = "Average single-cell profile", 
     main = paste0("Shalek et al. (2013)"))
lines(smDat$x, smDat$pred, lwd = 3)
points(smDat$x, smDat$y, col = 2, lwd = 6, pch = 1)
 
fit <- lm( rowMeans(yy)~ xx)
l <- predict(fit, newdata = data.frame(x= xx))

zz <- ifelse(rowMeans(yy) < 0.5, 0, 1)
fit <- glm(zz ~ xx, family = binomial) # estimate bulk on log2 scale
y1 <- predict(fit, type = "response")
tmp <- y1

lines(sort(xx), y = tmp[order(xx)]*l[order(xx)], col = 4, lwd = 3)


#### Wu et al. (2014)
library(scRNASeqHCT116CellLinesWu)
data(eset_wu)

eset <- exprs(eset_wu)
fd <- fData(eset_wu)
pd <- pData(eset_wu)
xx = eset[,1];
yy = eset[,5:100]
smDat <- sm(x=xx, y =rowMeans(yy), n.bins = 25)
smDat <- smDat[-c(1:2),] 
plot(x = xx, y = rowMeans(yy), col = "gray", pch = ".", 
     xlim = c(0, 50), ylim = c(0, 50),  
     xlab = "Bulk profile", ylab = "Average single-cell profile", 
     main = paste0("Wu et al. (2014)"))
lines(smDat$x, smDat$pred, lwd = 3)
points(smDat$x, smDat$y, col = 2, lwd = 6, pch = 1)

fit <- lm( rowMeans(yy)~ xx)
l <- predict(fit, newdata = data.frame(x= xx))
zz <- ifelse(rowMeans(yy) < 4.5, 0, 1)
fit <- glm(zz ~ xx, family = binomial)
y1 <- predict(fit, type = "response")
tmp <- y1

lines(sort(xx), y = tmp[order(xx)]*l[order(xx)], col = 4, lwd = 3)


# Trapnell et al. (2014)
library(trapnell2014myoblasthuman)
data("trapnell2014myoblasthuman")

pd <- pData(trapnell2014myoblasthuman)
eset <- exprs(trapnell2014myoblasthuman)

# pick out only single cells and bulk cells; remove debris wells
pd <- pd[pd$debris == "debris: FALSE" &
         pd$numcells %in% c("cells in well: 1", 
                            "cells in well: ~200000"), ]
eset <- eset[, match(rownames(pd), colnames(eset))]
eset <- eset[rowSums(eset) > 0, ]

pd$batch <- factor(paste(pd$runID, pd$fcLane, sep = "_"))
pd$hour <- factor(pd$hour, levels = levels(pd$hour)[c(2,3,4,1)])
levels(pd$hour) <- (c("0h", "24h", "48h", "72h"))

tme = 4
esetSub <- eset[, pd$hour == levels(pd$hour)[tme]]
pdSub <- pd[pd$hour == levels(pd$hour)[tme], ]

esetBulk <- esetSub[, pdSub$sampleType == "bulk"]
esetSC <- esetSub[, pdSub$numcells == "cells in well: 1"]

xx = esetBulk[,1]
yy = esetSC
smDat <- sm(x=xx, y =rowMeans(yy), n.bins = 30)
plot(x = xx, y = rowMeans(yy), col = "gray", pch = ".", 
     xlim = c(0, 30), ylim = c(0, 30),  
     xlab = "Bulk profile", ylab = "Average single-cell profile", 
     main = paste0("Trapnell et al. (2014)"))
lines(smDat$x, smDat$pred, lwd = 3)
points(smDat$x, smDat$y, col = 2, lwd = 6, pch = 1)

fit <- lm( rowMeans(yy)~ xx)
l <- predict(fit, newdata = data.frame(x= xx))
zz <- ifelse(rowMeans(yy) < 0.5, 0, 1)
fit <- glm(zz ~ xx, family = binomial)
y1 <- predict(fit, type = "response")
tmp <- y1

lines(sort(xx), y = tmp[order(xx)]*l[order(xx)], col = 4, lwd = 3)

dev.off()

```

# Figure - Bulk vs SC ratio of obs / expected
```{r}
pdf(file.path(working_path, "Figure_scatter_ratio_log2scale.pdf"), width = 20, height =6) 
par(mfrow=c(1,3), mai=c(0.75,0.75,0.4,0.1), mgp=c(3.3,1.25,0), 
    cex.lab=2.25, cex.axis=2.25, cex.main=2.25, cex.sub=2.25)

#### Shalek et al. (2013)
library(scRNASeqMouseShalekBMDCs)
data("BMDCsMouseTPM")

eset <- exprs(BMDCsMouseTPM)
pd <- pData(BMDCsMouseTPM)

esetBulk <- eset[, pd$sampleType == "bulk"]
esetSC <- eset[, pd$sampleType == "SC"]

xx = esetBulk[,1];
yy = esetSC
smDat <- sm(x=xx, y =rowMeans(yy), n.bins = 50)
plot(log2(smDat$x), smDat$y / (smDat$pred), pch = 1,  xaxt= "n",
     xlim = c(-4, 7), ylim = c(0, 1.05), lwd = 5,
     xlab = "Bulk profile (TPM)", ylab = "Average single-cell profile (Observed/Expected)", 
     main = paste0("Shalek et al. (2013)"))
lab = 2^c(-4:7)
axis(1, at = log2(lab), labels = lab)

zz <- ifelse(rowMeans(yy) < 1, 0, 1)
fit <- glm(zz ~ log2(xx+1), family = binomial)
y1 <- predict(fit, type = "response")
tmp <- y1

lines(log2(sort(xx)), y = tmp[order(xx)] , col = 4, lwd = 3)


#### Wu et al. (2014)
library(scRNASeqHCT116CellLinesWu)
data(eset_wu)

eset <- exprs(eset_wu)
fd <- fData(eset_wu)
pd <- pData(eset_wu)
xx = eset[,1];
yy = eset[,5:100]
smDat <- sm(x=xx, y =rowMeans(yy), n.bins = 25)
smDat <- smDat[-c(1:2),] 
plot(log2(smDat$x), smDat$y / (smDat$pred), pch = 1, xaxt= "n",
     xlim = c(-4, 7), ylim = c(0, 1.05), lwd = 5,
     xlab = "Bulk profile (FPKM)", ylab = "Average single-cell profile (Observed/Expected)", 
     main = paste0("Wu et al. (2014)"))
lab = 2^c(-4:7)
axis(1, at = log2(lab), labels = lab)

zz <- ifelse(rowMeans(yy) < 4.5, 0, 1)
fit <- glm(zz ~ log2(xx+1), family = binomial)
y1 <- predict(fit, type = "response")
tmp <- y1

lines(log2(sort(xx)), y = tmp[order(xx)], col = 4, lwd = 3)





# Trapnell et al. (2014)
library(trapnell2014myoblasthuman)
data("trapnell2014myoblasthuman")

pd <- pData(trapnell2014myoblasthuman)
eset <- exprs(trapnell2014myoblasthuman)

# pick out only single cells and bulk cells; remove debris wells
pd <- pd[pd$debris == "debris: FALSE" &
         pd$numcells %in% c("cells in well: 1", 
                            "cells in well: ~200000"), ]
eset <- eset[, match(rownames(pd), colnames(eset))]
eset <- eset[rowSums(eset) > 0, ]

pd$batch <- factor(paste(pd$runID, pd$fcLane, sep = "_"))
pd$hour <- factor(pd$hour, levels = levels(pd$hour)[c(2,3,4,1)])
levels(pd$hour) <- (c("0h", "24h", "48h", "72h"))

tme = 4
esetSub <- eset[, pd$hour == levels(pd$hour)[tme]]
pdSub <- pd[pd$hour == levels(pd$hour)[tme], ]

esetBulk <- esetSub[, pdSub$sampleType == "bulk"]
esetSC <- esetSub[, pdSub$numcells == "cells in well: 1"]

xx = esetBulk[,1]
yy = esetSC
smDat <- sm(x=xx, y =rowMeans(yy), n.bins = 30)
plot(log2(smDat$x), smDat$y / (smDat$pred), pch = 1, xaxt = "n",
     xlim = c(-4, 7), ylim = c(0, 1.05), lwd = 5, 
     xlab = "Bulk profile (FPKM)", ylab = "Average single-cell profile (Observed/Expected)", 
     main = paste0("Trapnell et al. (2014)"))
lab = 2^c(-4:7)
axis(1, at = log2(lab), labels = lab)

zz <- ifelse(rowMeans(yy) < 0.5, 0, 1)
fit <- glm(zz ~ log2(xx+1), family = binomial)
y1 <- predict(fit, type = "response")
tmp <- y1

lines(log2(sort(xx)), y = tmp[order(xx)], col = 4, lwd = 3)

dev.off()

```


# Figure Bulk vs SC logistic curves estimated per cell
```{r}
par(mfrow=c(1,3), mai=c(0.75,0.75,0.4,0.1), mgp=c(3.3,1.25,0), 
    cex.lab=2.25, cex.axis=2.25, cex.main=2.25, cex.sub=2.25)

#### Shalek et al. (2013)
library(scRNASeqMouseShalekBMDCs)
data("BMDCsMouseTPM")

eset <- exprs(BMDCsMouseTPM)
pd <- pData(BMDCsMouseTPM)
pd$CDRlt1 <- colMeans(eset > 1)

esetBulk <- eset[, pd$sampleType == "bulk"]
esetSC <- eset[, pd$sampleType == "SC"]

xx = esetBulk[,1];
yy = esetSC

z <- 2^c(-4:16)
est.coeffs <- array(0, dim = c(dim(yy)[2], 2))
pred.y = array(0, dim = c(length(z), dim(yy)[2]))
for(i in 1:dim(yy)[2]){
  zz <- ifelse(esetSC[,i] > 1, 1, 0)
  fit = glm(zz ~ log2(xx+1), family = binomial)
  est.coeffs[i,] <- fit$coefficients
  pred.y[,i] <- locfit::expit(fit$coefficients[1] + fit$coefficients[2]*log2(z))
  print(i)
}

colnames(pred.y) <- pd[pd$sampleType == "SC", ]$geo_accession
dat <- data.frame(melt(pred.y), 
                  "bulk" = rep(log2(z), ncol(pred.y)))

p.probdetect.shalek <- ggplot(dat, aes(x = bulk, y = value, group = Var2)) + 
  geom_line(size = 1) + ylim(c(0, 1)) + 
  labs(title = "Shalek et al. (2013)") + 
  xlab("Bulk profile (log2 scale)") + ylab("Probability of detection") + 
       theme(plot.title = element_text(size = 25), 
          axis.text = element_text(size = 25), 
          axis.title = element_text(size = 25))

dat2 <- data.frame(pd[pd$sampleType == "SC", ], 
                   "intercept" = est.coeffs[, 1], "slope" = est.coeffs[,2])
p.intercept.shalek <- ggplot(dat2, aes(x = CDRlt1, y = intercept))  + geom_point(size = 3) + 
       xlab("Detection rate") + ylab("Intercept") + 
       theme(plot.title = element_text(size = 25), 
          axis.text = element_text(size = 25), 
          axis.title = element_text(size = 25))

p.slope.shalek <- ggplot(dat2, aes(x = slope))  + geom_histogram(bins=10) + 
       xlab("slope") + 
       theme(plot.title = element_text(size = 25), 
          axis.text = element_text(size = 25), 
          axis.title = element_text(size = 25))


#### Wu et al. (2014)
library(scRNASeqHCT116CellLinesWu)
data(eset_wu)

eset <- exprs(eset_wu)
fd <- fData(eset_wu)
pd <- pData(eset_wu)
pd$CDRlt1 <- colMeans(eset > 1)
eset <- eset[, pd$CDRlt1 > 0.1]
pd <- pd[pd$CDRlt1 > 0.1, ]
xx = eset[,1];
yy = eset[,5:nrow(pd)]

z <- 2^c(-4:16)
est.coeffs <- array(0, dim = c(dim(yy)[2], 2))
pred.y = array(0, dim = c(length(z), dim(yy)[2]))
for(i in 1:dim(yy)[2]){
  zz <- ifelse(eset[,4+i] > 1, 1, 0)
  fit = glm(zz ~ log2(xx+1), family = binomial)
  est.coeffs[i,] <- fit$coefficients
  pred.y[,i] <- locfit::expit(fit$coefficients[1] + fit$coefficients[2]*log2(z))
  print(i)
}

colnames(pred.y) <- pd[5:nrow(pd), ]$Sample_geo_accession
dat <- data.frame(melt(pred.y), 
                  "bulk" = rep(log2(z), ncol(pred.y)))

p.probdetect.wu <- ggplot(dat, aes(x = bulk, y = value, group = Var2)) + 
  geom_line(size = 1) + ylim(c(0, 1)) + 
  labs(title = "Wu et al. (2014)") + 
  xlab("Bulk profile (log2 scale)") + ylab("Probability of detection") + 
       theme(plot.title = element_text(size = 25), 
          axis.text = element_text(size = 25), 
          axis.title = element_text(size = 25))

dat2 <- data.frame(pd[5:nrow(pd), ], "intercept" = est.coeffs[, 1], "slope" = est.coeffs[,2])
p.intercept.wu <- ggplot(dat2, aes(x = CDRlt1, y = intercept))  + geom_point(size = 3) + 
       xlab("Detection rate") + ylab("Intercept") + 
       theme(plot.title = element_text(size = 25), 
          axis.text = element_text(size = 25), 
          axis.title = element_text(size = 25))

p.slope.wu <- ggplot(dat2, aes(x = slope))  + geom_histogram(bins=20) +  
       xlab("slope") + 
       theme(plot.title = element_text(size = 25), 
          axis.text = element_text(size = 25), 
          axis.title = element_text(size = 25))


# Trapnell et al. (2014)
library(trapnell2014myoblasthuman)
data("trapnell2014myoblasthuman")

pd <- pData(trapnell2014myoblasthuman)
eset <- exprs(trapnell2014myoblasthuman)

# pick out only single cells and bulk cells; remove debris wells
pd <- pd[pd$debris == "debris: FALSE" &
         pd$numcells %in% c("cells in well: 1", 
                            "cells in well: ~200000"), ]
eset <- eset[, match(rownames(pd), colnames(eset))]
eset <- eset[rowSums(eset) > 0, ]

pd$batch <- factor(paste(pd$runID, pd$fcLane, sep = "_"))
pd$hour <- factor(pd$hour, levels = levels(pd$hour)[c(2,3,4,1)])
levels(pd$hour) <- (c("0h", "24h", "48h", "72h"))

pd$CDRlt1 <- colMeans(eset > 1)

tme = 4
esetSub <- eset[, pd$hour == levels(pd$hour)[tme]]
pdSub <- pd[pd$hour == levels(pd$hour)[tme], ]

esetBulk <- esetSub[, pdSub$sampleType == "bulk"]
esetSC <- esetSub[, pdSub$numcells == "cells in well: 1"]
esetSC <- esetSC[, pdSub[pdSub$numcells == "cells in well: 1", ]$CDRlt1 > 0.05]
pdSub <- pdSub[pdSub$CDRlt1 > 0.05, ]

xx = esetBulk[,1]
yy = esetSC

z <- 2^c(-4:16)
est.coeffs <- array(0, dim = c(dim(yy)[2], 2))
pred.y = array(0, dim = c(length(z), dim(yy)[2]))
for(i in 1:dim(yy)[2]){
  zz <- ifelse(esetSC[, i] > 1, 1, 0)
  fit = glm(zz ~ log2(xx+1), family = binomial)
  est.coeffs[i,] <- fit$coefficients
  pred.y[,i] <- locfit::expit(fit$coefficients[1] + fit$coefficients[2]*log2(z))
  print(i)
}

colnames(pred.y) <- pd[pd$numcells == "cells in well: 1" & 
                         pd$CDRlt1 > 0.05 & 
                         pd$hour == levels(pd$hour)[tme], ]$geo_accession
dat <- data.frame(melt(pred.y), 
                  "bulk" = rep(log2(z), ncol(pred.y)))

p.probdetect.trapnell <- ggplot(dat, aes(x = bulk, y = value, group = Var2)) + 
  geom_line(size = 1) + ylim(c(0, 1)) + 
  labs(title = "Trapnell et al. (2014)") + 
  xlab("Bulk profile (log2 scale)") + ylab("Probability of detection") + 
       theme(plot.title = element_text(size = 25), 
          axis.text = element_text(size = 25), 
          axis.title = element_text(size = 25))

dat2 <- data.frame(pd[pd$numcells == "cells in well: 1" & 
                        pd$CDRlt1 > 0.05 & pd$hour == levels(pd$hour)[tme], ], 
                   "intercept" = est.coeffs[, 1], "slope" = est.coeffs[,2])
p.intercept.trapnell <- ggplot(dat2, aes(x = CDRlt1, y = intercept))  + geom_point(size = 3) + 
       xlab("Detection rate") + ylab("Intercept") + 
       theme(plot.title = element_text(size = 25), 
          axis.text = element_text(size = 25), 
          axis.title = element_text(size = 25))

p.slope.trapnell <- ggplot(dat2, aes(x=slope))  + geom_histogram(bins=20) +
       xlab("slope") + 
       theme(plot.title = element_text(size = 25), 
          axis.text = element_text(size = 25), 
          axis.title = element_text(size = 25))



p.out <- plot_grid(p.probdetect.shalek, p.probdetect.wu, p.probdetect.trapnell, 
                   p.intercept.shalek, p.intercept.wu, p.intercept.trapnell, 
                   labels = LETTERS[1:6], ncol = 3, label_size = 25)
pdf(file.path(working_path, "Figure_scatter_logisticPerCell.pdf"), width = 16, height = 10)
print(p.out)
dev.off()

p.out <- plot_grid(p.slope.shalek, p.slope.wu, p.slope.trapnell, 
                   labels = LETTERS[1:3], ncol = 3, label_size = 25)
pdf(file.path(working_path, "Figure_scatter_logisticPerCell-slopes.pdf"), width = 16, height = 5)
print(p.out)
dev.off()


```


# Figure Bulk vs SC reproduce scatter and MA plots

```{r}
pdf(file.path(working_path, "Figure_scatter_reproduce.pdf"), width = 22, height =16) 
par(mfrow=c(2,3), mai=c(0.75,0.75,0.4,0.1), mgp=c(3.3,1.25,0),
    cex.lab=2.25, cex.axis=2.25, cex.main=2.25, cex.sub=2.25)

#### Shalek et al. (2013)
library(scRNASeqMouseShalekBMDCs)
data("BMDCsMouseTPM")

eset <- exprs(BMDCsMouseTPM)
pd <- pData(BMDCsMouseTPM)

esetBulk <- eset[, pd$sampleType == "bulk"]
esetSC <- eset[, pd$sampleType == "SC"]

xx = esetBulk[,1];
yy = esetSC
plot(x = log2(xx+1), y = log2(rowMeans(yy)+1), col = "black", pch = ".",
     xlim = c(0, 15), ylim = c(0, 15),
     xlab = expression("log"[2]*"(bulk profile)"),
     ylab = expression("log"[2]*"(single-cell profile average)"),
     main = paste0("Shalek et al. (2013)"))
abline(0, 1, col = 2, lwd = 3)


#### Wu et al. (2014)
library(scRNASeqHCT116CellLinesWu)
data(eset_wu)

eset <- exprs(eset_wu)
fd <- fData(eset_wu)
pd <- pData(eset_wu)

xx = eset[, 1] # eset[, 1]  # 
yy = eset[, 5:100] # eset[, 5:100] # 
plot(x = log2(xx+1), y = log2(rowMeans(yy)+1), col = "black", pch = ".",
     xlim = c(0, 15), ylim = c(0, 15),
     xlab = expression("log"[2]*"(bulk profile)"),
     ylab = expression("log"[2]*"(single-cell profile average)"),
     main = paste0("Wu et al. (2014)"))
abline(0, 1, col = 2, lwd = 3)

# Trapnell
library(trapnell2014myoblasthuman)
data("trapnell2014myoblasthuman")

pd <- pData(trapnell2014myoblasthuman)
eset <- exprs(trapnell2014myoblasthuman)

# pick out only single cells and bulk cells; remove debris wells
pd <- pd[pd$debris == "debris: FALSE" &
         pd$numcells %in% c("cells in well: 1", 
                            "cells in well: ~200000"), ]
eset <- eset[, match(rownames(pd), colnames(eset))]
eset <- eset[rowSums(eset) > 0, ]

pd$batch <- factor(paste(pd$runID, pd$fcLane, sep = "_"))
pd$hour <- factor(pd$hour, levels = levels(pd$hour)[c(2,3,4,1)])
levels(pd$hour) <- (c("0h", "24h", "48h", "72h"))

tme = 4
esetSub <- eset[, pd$hour == levels(pd$hour)[tme]]
pdSub <- pd[pd$hour == levels(pd$hour)[tme], ]

esetBulk <- esetSub[, pdSub$sampleType == "bulk"]
esetSC <- esetSub[, pdSub$numcells == "cells in well: 1"]

xx = esetBulk[,1]
yy = esetSC
plot(x = log2(xx+1), y = log2(rowMeans(yy)+1), col = "black", pch = ".",
     xlim = c(0, 15), ylim = c(0, 15),
     xlab = expression("log"[2]*"(bulk profile)"),
     ylab = expression("log"[2]*"(single-cell profile average)"),
     main = paste0("Trapnell et al. (2014)"))
abline(0, 1, col = 2, lwd = 3)




#########################  ma plot (Row 2)

#### Shalek et al. (2013)
library(scRNASeqMouseShalekBMDCs)
data("BMDCsMouseTPM")

eset <- exprs(BMDCsMouseTPM)
pd <- pData(BMDCsMouseTPM)

esetBulk <- eset[, pd$sampleType == "bulk"]
esetSC <- eset[, pd$sampleType == "SC"]

xx = (log2(esetBulk[,1]+1) + log2(rowMeans(esetSC)+1))/2
yy = log2(rowMeans(esetSC)+1) - log2(esetBulk[,1]+1)
plot(x = xx, y = yy, ylim = c(-5, 5), pch = ".",
       xlab = expression("A = (log"[2]*"(single-cell profile avg) + log"[2]*"(bulk profile))/2"),
       ylab = expression("M = log"[2]*"(single-cell profile avg) - log"[2]*"(bulk profile)"),
       main = paste0("Shalek et al. (2013)"))
  abline(h = 0, col = 2, lwd = 3)
  smDat <- sm(x = xx, y = yy)
  lines(smDat$x, smDat$y, lwd = 5, col = 4)

 
 
#### Wu et al. (2014)
library(scRNASeqHCT116CellLinesWu)
data(eset_wu)

eset <- exprs(eset_wu)
fd <- fData(eset_wu)
pd <- pData(eset_wu)
i = 1
xx = (log2(c(rowMeans(eset[,5:100])+1)) + log2(c(eset[,i]+1)))/2
yy = log2(rowMeans(eset[,5:100])+1) - log2(eset[,i]+1)
plot(x = xx,y = yy, pch = ".", ylim = c(-5, 5), 
       xlab = expression("A = (log"[2]*"(single-cell profile avg) + log"[2]*"(bulk profile))/2"),
       ylab = expression("M = log"[2]*"(single-cell profile avg) - log"[2]*"(bulk profile)"),
     main = paste0("Wu et al. (2014)"))
abline(h=0, col = 2, lwd = 3)
smDat <- sm(x=xx, y =yy)
lines(smDat$x, smDat$y, lwd = 5, col = 4)


# Trapnell
library(trapnell2014myoblasthuman)
data("trapnell2014myoblasthuman")

pd <- pData(trapnell2014myoblasthuman)
eset <- exprs(trapnell2014myoblasthuman)

# pick out only single cells and bulk cells; remove debris wells
pd <- pd[pd$debris == "debris: FALSE" &
         pd$numcells %in% c("cells in well: 1", 
                            "cells in well: ~200000"), ]
eset <- eset[, match(rownames(pd), colnames(eset))]
eset <- eset[rowSums(eset) > 0, ]

pd$batch <- factor(paste(pd$runID, pd$fcLane, sep = "_"))
pd$hour <- factor(pd$hour, levels = levels(pd$hour)[c(2,3,4,1)])
levels(pd$hour) <- (c("0h", "24h", "48h", "72h"))

tme = 4
esetSub <- eset[, pd$hour == levels(pd$hour)[tme]]
pdSub <- pd[pd$hour == levels(pd$hour)[tme], ]

esetBulk <- esetSub[, pdSub$sampleType == "bulk"]
esetSC <- esetSub[, pdSub$numcells == "cells in well: 1"]

xx = (log2(esetBulk[,1]+1) + log2(rowMeans(esetSC)+1))/2
yy = log2(rowMeans(esetSC)+1) - log2(esetBulk[,1]+1)
plot(x = xx, y = yy, ylim = c(-5, 5), 
       xlab = expression("A = (log"[2]*"(single-cell profile avg) + log"[2]*"(bulk profile))/2"),
       ylab = expression("M = log"[2]*"(single-cell profile avg) - log"[2]*"(bulk profile)"),
       main = paste0("Trapnell et al. (2014)"), pch = ".")
abline(h = 0, col = 2, lwd = 3)
smDat <- sm(x=xx, y =yy)
lines(smDat$x, smDat$y, lwd = 5, col = 4)

dev.off()


```


```{r}
sessionInfo()
```
